<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:w="urn:schemas-microsoft-com:office:word"
xmlns:m="http://schemas.microsoft.com/office/2004/12/omml"
xmlns:css="http://macVmlSchemaUri" xmlns="http://www.w3.org/TR/REC-html40">

<head>
<meta name=Title content="GISTIC Documentation">
<meta name=Keywords content="">
<meta http-equiv=Content-Type content="text/html; charset=windows-1252">
<meta name=ProgId content=Word.Document>
<meta name=Generator content="Microsoft Word 2008">
<meta name=Originator content="Microsoft Word 2008">
<base target="_blank">
<link rel=File-List href="GISTICDocumentation_standalone_files/filelist.xml">
<link rel=Edit-Time-Data
href="GISTICDocumentation_standalone_files/editdata.mso">
<!--[if !mso]>
<style>
v\:* {behavior:url(#default#VML);}
o\:* {behavior:url(#default#VML);}
w\:* {behavior:url(#default#VML);}
.shape {behavior:url(#default#VML);}
</style>
<![endif]-->
<title>GISTIC Documentation</title>
<!--[if gte mso 9]><xml>
 <o:DocumentProperties>
  <o:Author>Jennifer Dobson</o:Author>
  <o:Template>Normal.dotm</o:Template>
  <o:LastAuthor>Steven Schumacher</o:LastAuthor>
  <o:Revision>9</o:Revision>
  <o:TotalTime>93</o:TotalTime>
  <o:Created>2011-03-24T18:43:00Z</o:Created>
  <o:LastSaved>2012-05-08T14:24:00Z</o:LastSaved>
  <o:Pages>4</o:Pages>
  <o:Words>2032</o:Words>
  <o:Characters>11584</o:Characters>
  <o:Company>CGR/MIT</o:Company>
  <o:Lines>96</o:Lines>
  <o:Paragraphs>23</o:Paragraphs>
  <o:CharactersWithSpaces>14225</o:CharactersWithSpaces>
  <o:Version>12.0</o:Version>
 </o:DocumentProperties>
 <o:OfficeDocumentSettings>
  <o:AllowPNG/>
 </o:OfficeDocumentSettings>
</xml><![endif]--><!--[if gte mso 9]><xml>
 <w:WordDocument>
  <w:Zoom>150</w:Zoom>
  <w:TrackMoves>false</w:TrackMoves>
  <w:TrackFormatting/>
  <w:ValidateAgainstSchemas/>
  <w:SaveIfXMLInvalid>false</w:SaveIfXMLInvalid>
  <w:IgnoreMixedContent>false</w:IgnoreMixedContent>
  <w:AlwaysShowPlaceholderText>false</w:AlwaysShowPlaceholderText>
  <w:Compatibility>
   <w:BreakWrappedTables/>
   <w:SplitPgBreakAndParaMark/>
   <w:UseFELayout/>
  </w:Compatibility>
 </w:WordDocument>
</xml><![endif]--><!--[if gte mso 9]><xml>
 <w:LatentStyles DefLockedState="false" LatentStyleCount="276">
 </w:LatentStyles>
</xml><![endif]-->
<style>
<!--table.MSONORMALTABLE
	{mso-bidi-language:"\#0400";}
table.MSOTABLEGRID
	{mso-bidi-language:"\#0400";}

 /* Font Definitions */
@font-face
	{font-family:Arial;
	panose-1:2 11 6 4 2 2 2 2 2 4;
	mso-font-charset:0;
	mso-generic-font-family:auto;
	mso-font-pitch:variable;
	mso-font-signature:3 0 0 0 1 0;}
@font-face
	{font-family:Times;
	panose-1:2 0 5 0 0 0 0 0 0 0;
	mso-font-charset:0;
	mso-generic-font-family:auto;
	mso-font-pitch:variable;
	mso-font-signature:3 0 0 0 1 0;}
@font-face
	{font-family:Cambria;
	panose-1:2 4 5 3 5 4 6 3 2 4;
	mso-font-charset:0;
	mso-generic-font-family:auto;
	mso-font-pitch:variable;
	mso-font-signature:3 0 0 0 1 0;}
@font-face
	{font-family:"Arial Black";
	panose-1:2 11 10 4 2 1 2 2 2 4;
	mso-font-charset:0;
	mso-generic-font-family:auto;
	mso-font-pitch:variable;
	mso-font-signature:3 0 0 0 1 0;}
 /* Style Definitions */
p.MsoNormal, li.MsoNormal, div.MsoNormal
	{mso-style-parent:"";
	margin:0in;
	margin-bottom:.0001pt;
	mso-pagination:widow-orphan;
	font-size:12.0pt;
	font-family:"Times New Roman";
	mso-fareast-font-family:"Times New Roman";
	mso-bidi-font-family:"Times New Roman";
	mso-bidi-theme-font:minor-bidi;}
a:link, span.MsoHyperlink
	{color:blue;
	text-decoration:underline;
	text-underline:single;}
a:visited, span.MsoHyperlinkFollowed
	{color:purple;
	text-decoration:underline;
	text-underline:single;}
p.msonormal1, li.msonormal1, div.msonormal1
	{mso-style-name:msonormal1;
	margin:0in;
	margin-bottom:.0001pt;
	mso-pagination:widow-orphan;
	font-size:10.0pt;
	font-family:"Times New Roman";
	mso-ascii-font-family:Times;
	mso-fareast-font-family:"Times New Roman";
	mso-fareast-theme-font:minor-fareast;
	mso-hansi-font-family:Times;
	mso-bidi-font-family:"Times New Roman";
	mso-bidi-theme-font:minor-bidi;}
@page Section1
	{size:8.5in 11.0in;
	margin:1.0in 1.25in 1.0in 1.25in;
	mso-header-margin:.5in;
	mso-footer-margin:.5in;
	mso-paper-source:0;}
div.Section1
	{page:Section1;}
 /* List Definitions */
@list l0
	{mso-list-id:351995387;
	mso-list-type:hybrid;
	mso-list-template-ids:-55140292 652265324 67698691 67698693 67698689 67698691 67698693 67698689 67698691 67698693;}
@list l0:level1
	{mso-level-start-at:0;
	mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:.75in;
	mso-level-number-position:left;
	margin-left:.75in;
	text-indent:-.25in;
	font-family:Symbol;
	mso-fareast-font-family:"Times New Roman";
	mso-bidi-font-family:Arial;}
@list l0:level2
	{mso-level-tab-stop:1.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l0:level3
	{mso-level-tab-stop:1.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l0:level4
	{mso-level-tab-stop:2.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l0:level5
	{mso-level-tab-stop:2.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l0:level6
	{mso-level-tab-stop:3.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l0:level7
	{mso-level-tab-stop:3.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l0:level8
	{mso-level-tab-stop:4.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l0:level9
	{mso-level-tab-stop:4.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l1
	{mso-list-id:781001619;
	mso-list-type:hybrid;
	mso-list-template-ids:679480794 650424718 67698713 67698715 67698703 67698713 67698715 67698703 67698713 67698715;}
@list l1:level1
	{mso-level-text:"\(%1\)";
	mso-level-tab-stop:34.5pt;
	mso-level-number-position:left;
	margin-left:34.5pt;
	text-indent:-.25in;
	mso-ansi-font-style:normal;}
@list l1:level2
	{mso-level-tab-stop:1.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l1:level3
	{mso-level-tab-stop:1.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l1:level4
	{mso-level-tab-stop:2.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l1:level5
	{mso-level-tab-stop:2.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l1:level6
	{mso-level-tab-stop:3.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l1:level7
	{mso-level-tab-stop:3.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l1:level8
	{mso-level-tab-stop:4.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l1:level9
	{mso-level-tab-stop:4.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l2
	{mso-list-id:1345550351;
	mso-list-type:hybrid;
	mso-list-template-ids:-429349394 1390080440 67698713 67698715 67698703 67698713 67698715 67698703 67698713 67698715;}
@list l2:level1
	{mso-level-text:"\(%1\)";
	mso-level-tab-stop:27.75pt;
	mso-level-number-position:left;
	margin-left:27.75pt;
	text-indent:-.25in;}
@list l2:level2
	{mso-level-tab-stop:1.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l2:level3
	{mso-level-tab-stop:1.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l2:level4
	{mso-level-tab-stop:2.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l2:level5
	{mso-level-tab-stop:2.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l2:level6
	{mso-level-tab-stop:3.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l2:level7
	{mso-level-tab-stop:3.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l2:level8
	{mso-level-tab-stop:4.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l2:level9
	{mso-level-tab-stop:4.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l3
	{mso-list-id:1426461155;
	mso-list-type:hybrid;
	mso-list-template-ids:1846677758 -45437028 67698713 67698715 67698703 67698713 67698715 67698703 67698713 67698715;}
@list l3:level1
	{mso-level-text:"\(%1\)";
	mso-level-tab-stop:37.5pt;
	mso-level-number-position:left;
	margin-left:37.5pt;
	text-indent:-.25in;}
@list l3:level2
	{mso-level-tab-stop:1.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l3:level3
	{mso-level-tab-stop:1.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l3:level4
	{mso-level-tab-stop:2.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l3:level5
	{mso-level-tab-stop:2.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l3:level6
	{mso-level-tab-stop:3.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l3:level7
	{mso-level-tab-stop:3.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l3:level8
	{mso-level-tab-stop:4.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l3:level9
	{mso-level-tab-stop:4.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l4
	{mso-list-id:1447508692;
	mso-list-type:hybrid;
	mso-list-template-ids:1129368824 -1233850946 67698713 67698715 67698703 67698713 67698715 67698703 67698713 67698715;}
@list l4:level1
	{mso-level-text:"\(%1\)";
	mso-level-tab-stop:37.5pt;
	mso-level-number-position:left;
	margin-left:37.5pt;
	text-indent:-.25in;
	mso-ansi-font-style:normal;}
@list l4:level2
	{mso-level-tab-stop:1.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l4:level3
	{mso-level-tab-stop:1.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l4:level4
	{mso-level-tab-stop:2.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l4:level5
	{mso-level-tab-stop:2.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l4:level6
	{mso-level-tab-stop:3.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l4:level7
	{mso-level-tab-stop:3.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l4:level8
	{mso-level-tab-stop:4.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l4:level9
	{mso-level-tab-stop:4.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l5
	{mso-list-id:1752192324;
	mso-list-type:hybrid;
	mso-list-template-ids:447751710 263351578 67698713 67698715 67698703 67698713 67698715 67698703 67698713 67698715;}
@list l5:level1
	{mso-level-text:"\(%1\)";
	mso-level-tab-stop:27.75pt;
	mso-level-number-position:left;
	margin-left:27.75pt;
	text-indent:-.25in;}
@list l5:level2
	{mso-level-tab-stop:1.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l5:level3
	{mso-level-tab-stop:1.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l5:level4
	{mso-level-tab-stop:2.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l5:level5
	{mso-level-tab-stop:2.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l5:level6
	{mso-level-tab-stop:3.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l5:level7
	{mso-level-tab-stop:3.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l5:level8
	{mso-level-tab-stop:4.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l5:level9
	{mso-level-tab-stop:4.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l6
	{mso-list-id:1834681042;
	mso-list-type:hybrid;
	mso-list-template-ids:1177476104 -1417618940 67698713 67698715 67698703 67698713 67698715 67698703 67698713 67698715;}
@list l6:level1
	{mso-level-text:"\(%1\)";
	mso-level-tab-stop:1.25in;
	mso-level-number-position:left;
	margin-left:1.25in;
	text-indent:-.25in;}
@list l6:level2
	{mso-level-tab-stop:1.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l6:level3
	{mso-level-tab-stop:1.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l6:level4
	{mso-level-tab-stop:2.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l6:level5
	{mso-level-tab-stop:2.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l6:level6
	{mso-level-tab-stop:3.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l6:level7
	{mso-level-tab-stop:3.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l6:level8
	{mso-level-tab-stop:4.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l6:level9
	{mso-level-tab-stop:4.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l7
	{mso-list-id:2083792730;
	mso-list-type:hybrid;
	mso-list-template-ids:971267128 67698703 67698713 67698715 67698703 67698713 67698715 67698703 67698713 67698715;}
@list l7:level1
	{mso-level-tab-stop:.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l7:level2
	{mso-level-tab-stop:1.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l7:level3
	{mso-level-tab-stop:1.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l7:level4
	{mso-level-tab-stop:2.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l7:level5
	{mso-level-tab-stop:2.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l7:level6
	{mso-level-tab-stop:3.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l7:level7
	{mso-level-tab-stop:3.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l7:level8
	{mso-level-tab-stop:4.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l7:level9
	{mso-level-tab-stop:4.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
ol
	{margin-bottom:0in;}
ul
	{margin-bottom:0in;}
-->
</style>
<!--[if gte mso 10]>
<style>
 /* Style Definitions */
table.MsoNormalTable
	{mso-style-name:"Table Normal";
	mso-tstyle-rowband-size:0;
	mso-tstyle-colband-size:0;
	mso-style-noshow:yes;
	mso-style-parent:"";
	mso-padding-alt:0in 5.4pt 0in 5.4pt;
	mso-para-margin:0in;
	mso-para-margin-bottom:.0001pt;
	mso-pagination:widow-orphan;
	font-size:12.0pt;
	font-family:"Times New Roman";
	mso-ascii-font-family:Cambria;
	mso-ascii-theme-font:minor-latin;
	mso-hansi-font-family:Cambria;
	mso-hansi-theme-font:minor-latin;}
table.MsoTableGrid
	{mso-style-name:"Table Grid";
	mso-tstyle-rowband-size:0;
	mso-tstyle-colband-size:0;
	border:solid windowtext 1.0pt;
	mso-border-alt:solid windowtext .5pt;
	mso-padding-alt:0in 0in 0in 0in;
	mso-border-insideh:.5pt solid windowtext;
	mso-border-insidev:.5pt solid windowtext;
	mso-para-margin:0in;
	mso-para-margin-bottom:.0001pt;
	mso-pagination:widow-orphan;
	font-size:10.0pt;
	font-family:"Times New Roman";}
</style>
<![endif]--><!--[if gte mso 9]><xml>
 <o:shapedefaults v:ext="edit" spidmax="1112"/>
</xml><![endif]--><!--[if gte mso 9]><xml>
 <o:shapelayout v:ext="edit">
  <o:idmap v:ext="edit" data="1"/>
  <o:regrouptable v:ext="edit">
   <o:entry new="1" old="0"/>
  </o:regrouptable>
 </o:shapelayout></xml><![endif]-->
</head>

<body lang=EN-US link=blue vlink=purple style='tab-interval:.5in'>

<div class=Section1>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>

GISTIC2 Documentation

</b></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal;mso-bidi-font-weight:
normal'><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></b></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal;mso-bidi-font-weight:
normal'><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>
Module Name:
</span></b><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><span
style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>
GISTIC2
</p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal;mso-bidi-font-weight:
normal'><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></b></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal;mso-bidi-font-weight:
normal'><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>Description:</span></b><span
style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'> <span
style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>Genomic
Identification of Significant Targets in Cancer, version 2.0 </p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal;mso-bidi-font-weight:
normal'><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></b></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal;mso-bidi-font-weight:
normal'><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>Authors:<span
style='mso-tab-count:2'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span></span></b><span
style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>Gad Getz,
Rameen Beroukhim, Craig Mermel, Steve Schumacher and Jen Dobson </p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal;mso-bidi-font-weight:
normal'><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></b></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal;mso-bidi-font-weight:
normal'><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>
Date:
<span style='mso-tab-count:3'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span></span></b><span
style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>
27 Mar 2017
</p>

<p class=MsoNormal><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal;mso-bidi-font-weight:
normal'><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>Release:</span></b><span
style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><span
style='mso-tab-count:2'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>2.0.23</p>

<p class=MsoNormal><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal;mso-bidi-font-weight:
normal'><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>Summary:</span></b><span
style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'></p>

<p class=MsoNormal>&nbsp;</p>

<p class=MsoNormal><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>The
GISTIC module identifies regions of the genome that are significantly amplified
or deleted across a set of samples.<span style="mso-spacerun: yes">&nbsp;
</span>Each aberration is assigned a G-score that considers the amplitude of
the aberration as well as the frequency of its occurrence across samples.<span
style="mso-spacerun: yes">&nbsp; </span>False Discovery Rate q-values are then
calculated for the aberrant regions, and regions with q-values below a
user-defined threshold are considered significant.<span style="mso-spacerun:
yes">&nbsp; </span>For each significant region, a “peak region” is identified,
which is the part of the aberrant region with greatest amplitude and frequency
of alteration.<span style="mso-spacerun: yes">&nbsp; </span>In addition, a
“wide peak” is determined using a leave-one-out algorithm to allow for errors
in the boundaries in a single sample. The “wide peak” boundaries are more
robust for identifying the most likely gene targets in the region. Each
significantly aberrant region is also tested to determine whether it results
primarily from broad events (longer than half a chromosome arm), focal events,
or significant levels of both.<span style="mso-spacerun: yes">&nbsp; </span>The
GISTIC module reports the genomic locations and calculated q-values for the
aberrant regions.<span style="mso-spacerun: yes">&nbsp; </span>It identifies
the samples that exhibit each significant amplification or deletion, and it
lists genes found in each “wide peak” region. </p>

<p class=MsoNormal><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal;mso-bidi-font-weight:
normal'><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>References:</span></b><span
style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'></p>

<p class=MsoNormal><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-left:.75in;text-indent:-.25in;mso-list:l0 level1 lfo1;
tab-stops:list .75in'><![if !supportLists]><span style='font-family:Symbol;
mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>·<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span style='font-family:Arial;mso-bidi-font-family:
"Times New Roman"'>Mermel C, Schumacher S, et al. (2011). &quot;GISTIC2.0
facilitates sensitive and confident localization of the targets of focal
somatic copy-number alteration in human cancers.&quot; <i style='mso-bidi-font-style:
normal;mso-bidi-font-style:normal'>Genome Biology</i>, <b style='mso-bidi-font-weight:
normal;mso-bidi-font-weight:normal'>12</b>:R41.</p>

<p class=MsoNormal style='margin-left:.75in;text-indent:-.25in;mso-list:l0 level1 lfo1;
tab-stops:list .75in'><![if !supportLists]><span style='font-family:Symbol;
mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>·<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span style='font-family:Arial;mso-bidi-font-family:
"Times New Roman"'>Beroukhim R, Mermel C, et al. (2010). &quot;The landscape of
somatic copy -number alteration across human cancers.&quot; <i
style='mso-bidi-font-style:normal;mso-bidi-font-style:normal'>Nature</i>, <b
style='mso-bidi-font-weight:normal;mso-bidi-font-weight:normal'>463</b>:899-905.</p>

<p class=MsoNormal style='margin-left:.75in;text-indent:-.25in;mso-list:l0 level1 lfo1;
tab-stops:list .75in'><![if !supportLists]><span style='font-family:Symbol;
mso-fareast-font-family:Symbol;mso-bidi-font-family:Symbol'><span
style='mso-list:Ignore'>·<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span style='font-family:Arial;mso-bidi-font-family:
"Times New Roman"'>Beroukhim R, Getz G, et al. (2007).<span
style="mso-spacerun: yes">&nbsp; </span>“Assessing the significance of
chromosomal abberations in cancer: Methodology and application to glioma.” <span
style="mso-spacerun: yes">&nbsp;</span><i style='mso-bidi-font-style:normal;
mso-bidi-font-style:normal'>Proc Natl Acad Sci,</i> <b style='mso-bidi-font-weight:
normal;mso-bidi-font-weight:normal'>104</b>:20007-20012.</p>

<p class=MsoNormal><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal;mso-bidi-font-weight:
normal'><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></b></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal;mso-bidi-font-weight:
normal'><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>Input
Parameters</b></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal;mso-bidi-font-weight:
normal'><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></b></p>

<table class=MsoTableGrid border=0 cellspacing=0 cellpadding=0
 style='margin-left:.2in;border-collapse:collapse;mso-yfti-tbllook:480;
 mso-padding-alt:0in 5.4pt 0in 5.4pt'>
 <tr style='mso-yfti-irow:0;mso-yfti-firstrow:yes'>
  <td width=122 valign=top style='width:121.5pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><b style='mso-bidi-font-weight:normal;mso-bidi-font-weight:
  normal'><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>Variable Name
  </b></p>
  </td>
  <td width=75 valign=top style='width:75.1pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><b style='mso-bidi-font-weight:normal;mso-bidi-font-weight:
  normal'><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>Option
  String </b></p>
  </td>
  <td width=269 valign=top style='width:268.8pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><b style='mso-bidi-font-weight:normal;mso-bidi-font-weight:
  normal'><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>Description
  </b></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:1'>
  <td width=122 valign=top style='width:121.5pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><i>base_dir</i>
  </p>
  </td>
  <td width=75 valign=top style='width:75.1pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>-b
  </p>
  </td>
  <td width=269 valign=top style='width:268.8pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>The
  directory in which to save all output files. (<i>REQUIRED</i>)</p>
  </td>
 </tr>

 <tr style='mso-yfti-irow:1'>
  <td width=122 valign=top style='width:121.5pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><i>seg_file</i>
  </p>
  </td>
  <td width=75 valign=top style='width:75.1pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>-seg
  </p>
  </td>
  <td width=269 valign=top style='width:268.8pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>Path to segmented
  data input file (<i>REQUIRED</i>, see below for file description). </p>
  </td>
 </tr>

 <tr style='mso-yfti-irow:1'>
  <td width=122 valign=top style='width:121.5pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><i>refgene</i>
  </p>
  </td>
  <td width=75 valign=top style='width:75.1pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>-refgene
  </p>
  </td>
  <td width=269 valign=top style='width:268.8pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>Path to reference genome
  data input file (<i>REQUIRED</i>, see below for file description). </p>
  </td>
 </tr>

 <tr style='mso-yfti-irow:1'>
  <td width=122 valign=top style='width:121.5pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><i>markers</i> (file path)
  </p>
  </td>
  <td width=75 valign=top style='width:75.1pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>-mk
  </p>
  </td>
  <td width=269 valign=top style='width:268.8pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>Path to markers input file
  (<i>OPTIONAL</i>, but encouraged: see below for file description). </p>
  </td>
 </tr>

 <tr style='mso-yfti-irow:4'>
  <td width=122 valign=top style='width:121.5pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1 style='tab-stops:90.75pt'>
    <span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><i>markers</i> (numeric)
  </p>
  </td>
  <td width=75 valign=top style='width:75.1pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>-maxspace
  </p>
  </td>
  <td width=269 valign=top style='width:268.8pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>
  Maximum allowed spacing between pseudo-markers, in bases. Pseudo-markers are generated 
  when the markers file input is omitted.<span
  style="mso-spacerun: yes">&nbsp; </span>Segments that contain fewer than this
  number of markers are joined to the neighboring segment that is closest in
  copy number.<span style="mso-spacerun: yes">&nbsp; </span>(DEFAULT=10,00) </p>
  </td>
 </tr>

 <tr style='mso-yfti-irow:2'>
  <td width=122 valign=top style='width:121.5pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><i>t_amp</i>
  </p>
  </td>
  <td width=75 valign=top style='width:75.1pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>-ta
  </p>
  </td>
  <td width=269 valign=top style='width:268.8pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>Threshold
  for copy number amplifications.<span style="mso-spacerun: yes">&nbsp;
  </span>Regions with a copy number gain above this value are considered amplified.
  (DEFAULT=0.1) </p>
  </td>
 </tr>

 <tr style='mso-yfti-irow:3'>
  <td width=122 valign=top style='width:121.5pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1 style='tab-stops:90.75pt'><span style='font-family:Arial;
  mso-bidi-font-family:"Times New Roman"'><i>t_del</i></p>
  </td>
  <td width=75 valign=top style='width:75.1pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>-td
  </p>
  </td>
  <td width=269 valign=top style='width:268.8pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>Threshold
  for copy number deletions.<span style="mso-spacerun: yes">&nbsp;
  </span>Regions with a copy number loss below this value are
  considered deletions. (DEFAULT=0.1) </p>
  </td>
 </tr>

 <tr style='mso-yfti-irow:4'>
  <td width=122 valign=top style='width:121.5pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1 style='tab-stops:90.75pt'><span style='font-family:Arial;
  mso-bidi-font-family:"Times New Roman"'><i>join_segment_size</i>
  </p>
  </td>
  <td width=75 valign=top style='width:75.1pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>-js
  </p>
  </td>
  <td width=269 valign=top style='width:268.8pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>Smallest
  number of markers to allow in segments from the segmented data.<span
  style="mso-spacerun: yes">&nbsp; </span>Segments that contain fewer than this
  number of markers are joined to the neighboring segment that is closest in
  copy number.<span style="mso-spacerun: yes">&nbsp; </span>(DEFAULT=4) </p>
  </td>
 </tr>

 <tr style='mso-yfti-irow:5'>
  <td width=122 valign=top style='width:121.5pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1 style='tab-stops:90.75pt'><span style='font-family:Arial;
  mso-bidi-font-family:"Times New Roman"'><i>qv_thresh</i></p>
  </td>
  <td width=75 valign=top style='width:75.1pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>-qvt
  </p>
  </td>
  <td width=269 valign=top style='width:268.8pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>Significance
  threshold for q-values.<span style="mso-spacerun: yes">&nbsp; </span>Regions
  with q-values below this number are considered significant. (DEFAULT=0.25) </p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:6'>
  <td width=122 valign=top style='width:121.5pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1 style='tab-stops:90.75pt'><span style='font-family:Arial;
  mso-bidi-font-family:"Times New Roman"'><i>ext</i></p>
  </td>
  <td width=75 valign=top style='width:75.1pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>-ext
  </p>
  </td>
  <td width=269 valign=top style='width:268.8pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>Extension
  to append to all output files.<span style="mso-spacerun: yes">&nbsp;
  </span>(DEFAULT=’’, no extension) </p>
  </td>
 </tr>

 <tr style='mso-yfti-irow:7'>
  <td width=122 valign=top style='width:121.5pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1 style='tab-stops:90.75pt'><span style='font-family:Arial;
  mso-bidi-font-family:"Times New Roman"'><i>fname</i></p>
  </td>
  <td width=75 valign=top style='width:75.1pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>-fname
  </p>
  </td>
  <td width=269 valign=top style='width:268.8pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>Base
  file name to prepend to all output files.<span style="mso-spacerun:
  yes">&nbsp; </span>(DEFAULT no output basename) </p>
  </td>
 </tr>

 <tr style='mso-yfti-irow:8'>
  <td width=122 valign=top style='width:121.5pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1 style='tab-stops:90.75pt'><span style='font-family:Arial;
  mso-bidi-font-family:"Times New Roman"'><i>remove_X</i> </p>
  </td>
  <td width=75 valign=top style='width:75.1pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>-rx
  </p>
  </td>
  <td width=269 valign=top style='width:268.8pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>Flag
  indicating whether to remove data from the sex chromosomes before analysis.<span
  style="mso-spacerun: yes">&nbsp; </span>Allowed values= {1,0}. (DEFAULT=1,
  remove X,Y) </p>
  </td>
 </tr>

 <tr style='mso-yfti-irow:9'>
  <td width=122 valign=top style='width:121.5pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1 style='tab-stops:90.75pt'><span style='font-family:Arial;
  mso-bidi-font-family:"Times New Roman"'><i>cap</i> </p>
  </td>
  <td width=75 valign=top style='width:75.1pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>-cap
  </p>
  </td>
  <td width=269 valign=top style='width:268.8pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>Minimum
  and maximum cap values on analyzed data. Regions with a log2 ratio greater
  than the cap are set to the cap value; regions with a log2 ratio less than
  -cap value are set to -cap. Values must be positive. (DEFAULT=1.5) </p>
  </td>
 </tr>

 <tr style='mso-yfti-irow:10'>
  <td width=122 valign=top style='width:121.5pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1 style='tab-stops:90.75pt'><span style='font-family:Arial;
  mso-bidi-font-family:"Times New Roman"'><i>run_broad_analysis</i> </p>
  </td>
  <td width=75 valign=top style='width:75.1pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>-broad
  </p>
  </td>
  <td width=269 valign=top style='width:268.8pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>Flag
  indicating that an additional broad-level analysis should be performed.
  Allowed values = {1,0}. (DEFAULT = 0, no broad analysis). </p>
  </td>
 </tr>

 <tr style='mso-yfti-irow:11'>
  <td width=122 valign=top style='width:121.5pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1 style='tab-stops:90.75pt'><span style='font-family:Arial;
  mso-bidi-font-family:"Times New Roman"'><i>broad_len_cutoff</i> </p>
  </td>
  <td width=75 valign=top style='width:75.1pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>-brlen
  </p>
  </td>
  <td width=269 valign=top style='width:268.8pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>Threshold
  used to distinguish broad from focal events, given in units of fraction of chromosome
  arm. (DEFAULT = 0.98) </p>
  </td>
 </tr>

 <tr style='mso-yfti-irow:12'>
  <td width=122 valign=top style='width:121.5pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1 style='tab-stops:90.75pt'><span style='font-family:Arial;
  mso-bidi-font-family:"Times New Roman"'><i>use_two_sided</i></span><span
  style='font-family:"Times New Roman";mso-bidi-font-family:"Times New Roman"'>
  </p>
  </td>
  <td width=75 valign=top style='width:75.1pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>-twoside
  </p>
  </td>
  <td width=269 valign=top style='width:268.8pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>Flag
  indicating that a two-dimensional quadrant figure should be created as part
  of a broad analysis. Allowed values = {1,0}.(DEFAULT=0, no figure). </p>
  </td>
 </tr>

 <tr style='mso-yfti-irow:13'>
  <td width=122 valign=top style='width:121.5pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1 style='tab-stops:90.75pt'><span style='font-family:Arial;
  mso-bidi-font-family:"Times New Roman"'><i>ziggs.max_segs_per_sample</i></span><span
  style='font-family:"Times New Roman";mso-bidi-font-family:"Times New Roman"'>
  </p>
  </td>
  <td width=75 valign=top style='width:75.1pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>-maxseg
  </p>
  </td>
  <td width=269 valign=top style='width:268.8pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>Maximum
  number of segments allowed for a sample in the input data. Samples with more
  segments than this threshold are excluded from the analysis. (DEFAULT=2500) </p>
  </td>
 </tr>

 <tr style='mso-yfti-irow:14'>
  <td width=122 valign=top style='width:121.5pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1 style='tab-stops:90.75pt'><span style='font-family:Arial;
  mso-bidi-font-family:"Times New Roman"'><i>res</i></span><span
  style='font-family:"Times New Roman";mso-bidi-font-family:"Times New Roman"'>
  </p>
  </td>
  <td width=75 valign=top style='width:75.1pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>-res
  </p>
  </td>
  <td width=269 valign=top style='width:268.8pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>Resolution
  used to create the empirical distributions used to estimate background
  probabilities. Lower values generate more accurate results at a cost of
  greater computation time. (DEFAULT=0.05) </p>
  </td>
 </tr>

 <tr style='mso-yfti-irow:15'>
  <td width=122 valign=top style='width:121.5pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1 style='tab-stops:90.75pt'><span style='font-family:Arial;
  mso-bidi-font-family:"Times New Roman"'><i>conf_level</i></span><span
  style='font-family:"Times New Roman";mso-bidi-font-family:"Times New Roman"'>
  </p>
  </td>
  <td width=75 valign=top style='width:75.1pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>-conf
  </p>
  </td>
  <td width=269 valign=top style='width:268.8pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>Confidence
  level used to calculate the region containing a driver. (DEFAULT=0.75) </p>
  </td>
 </tr>

 <tr style='mso-yfti-irow:16'>
  <td width=122 valign=top style='width:121.5pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1 style='tab-stops:90.75pt'><span style='font-family:Arial;
  mso-bidi-font-family:"Times New Roman"'><i>do_gene_gistic</i></span><span
  style='font-family:"Times New Roman";mso-bidi-font-family:"Times New Roman"'>
  </p>
  </td>
  <td width=75 valign=top style='width:75.1pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>-genegistic
  </p>
  </td>
  <td width=269 valign=top style='width:268.8pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>Flag
  indicating that the gene GISTIC algorithm should be used to calculate the
  significance of deletions at a gene level instead of a marker level. Allowed
  values= {1,0}. (DEFAULT=0, no gene GISTIC). </p>
  </td>
 </tr>

 <tr style='mso-yfti-irow:17'>
  <td width=122 valign=top style='width:121.5pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1 style='tab-stops:90.75pt'><span style='font-family:Arial;
  mso-bidi-font-family:"Times New Roman"'><i>do_arbitration</i></span><span
  style='font-family:"Times New Roman";mso-bidi-font-family:"Times New Roman"'>
  </p>
  </td>
  <td width=75 valign=top style='width:75.1pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>-arb
  </p>
  </td>
  <td width=269 valign=top style='width:268.8pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>Flag
  for using the arbitrated peel-off algorithm when resolving the significance
  of overlapping peaks. Allowed values= {1,0}. (DEFAULT=1, use arbitrated
  peel-off) </p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:18'>
  <td width=122 valign=top style='width:121.5pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1 style='tab-stops:90.75pt'><span style='font-family:Arial;
  mso-bidi-font-family:"Times New Roman"'><i>peak_types</i></span><span
  style='font-family:"Times New Roman";mso-bidi-font-family:"Times New Roman"'>
  </p>
  </td>
  <td width=75 valign=top style='width:75.1pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>-peak_type
  </p>
  </td>
  <td width=269 valign=top style='width:268.8pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>Method
  for evaluating the significance of peaks, either <i style='mso-bidi-font-style:
  normal;mso-bidi-font-style:normal'>robust</i> (DEFAULT) or <i
  style='mso-bidi-font-style:normal;mso-bidi-font-style:normal'>loo</i> for
  leave-one-out. </p>
  </td>
 </tr>

 <tr style='mso-yfti-irow:19'>
  <td width=122 valign=top style='width:121.5pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1 style='tab-stops:90.75pt'><span style='font-family:Arial;
  mso-bidi-font-family:"Times New Roman"'><i>arm_peeloff</i></p>
  </td>
  <td width=75 valign=top style='width:75.1pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>-armpeel</p>
  </td>
  <td width=269 valign=top style='width:268.8pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>Flag
  set to enable arm-level peel-off of events during peak definition. The
  arm-level peel-off enhancement to the arbitrated peel-off method assigns all
  events in the same chromosome arm of the same sample to a single peak. It is
  useful when peaks are split by noise or chromothripsis. Allowed values=
  {1,0}. (DEFAULT=0, use normal arbitrated peel-off)</p>
  </td>
 </tr>

 <tr style='mso-yfti-irow:20'>
  <td width=122 valign=top style='width:121.5pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1 style='tab-stops:90.75pt'><span style='font-family:Arial;
  mso-bidi-font-family:"Times New Roman"'><i>sample_center</i></p>
  </td>
  <td width=75 valign=top style='width:75.1pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>-scent</p>
  </td>
  <td width=269 valign=top style='width:268.8pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>Method
  for centering each sample prior to the GISTIC analysis. Allowed values are median, 
  mean, or none. (DEFAULT=median) </p>
  </td>
 </tr>

 <tr style='mso-yfti-irow:21'>
  <td width=122 valign=top style='width:121.5pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1 style='tab-stops:90.75pt'><span style='font-family:Arial;
  mso-bidi-font-family:"Times New Roman"'><i>conserve_disk_space</i></span><span
  style='font-family:"Times New Roman";mso-bidi-font-family:"Times New Roman"'>
  </p>
  </td>
  <td width=75 valign=top style='width:75.1pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>-smalldisk
  </p>
  </td>
  <td width=269 valign=top style='width:268.8pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>Flag
  indicating that large <span style='font-variant:small-caps'>Matlab<sup>TM</sup></span>
  objects should not be saved to disk. Allowed values= {1,0}. (DEFAULT=0, save
  large). </p>
  </td>
 </tr>

 <tr style='mso-yfti-irow:22'>
  <td width=122 valign=top style='width:121.5pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1 style='tab-stops:90.75pt'><span style='font-family:Arial;
  mso-bidi-font-family:"Times New Roman"'><i>use_segarray</i></span><span
  style='font-family:"Times New Roman";mso-bidi-font-family:"Times New Roman"'>
  </p>
  </td>
  <td width=75 valign=top style='width:75.1pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>-smallmem
  </p>
  </td>
  <td width=269 valign=top style='width:268.8pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>Flag
  indicating that the SegArray memory compression scheme should be used to
  reduce the memory requirements of the computation for large data sets.
  Computation is somewhat slower with memory compression enabled. Allowed
  values= {1,0}. (DEFAULT=1, compress memory) </p>
  </td>
 </tr>

 <tr style='mso-yfti-irow:23'>
  <td width=122 valign=top style='width:121.5pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1 style='tab-stops:90.75pt'><span style='font-family:Arial;
  mso-bidi-font-family:"Times New Roman"'><i>write_gene_files</i> </p>
  </td>
  <td width=75 valign=top style='width:75.1pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>-savegene
  </p>
  </td>
  <td width=269 valign=top style='width:268.8pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>Flag
  indicating that gene tables should be saved. Allowed values= {1,0}.
  (DEFAULT=0, don't save gene tables)</span><span style='font-family:"Times New Roman";
  mso-bidi-font-family:"Times New Roman"'> </p>
  </td>
 </tr>

 <tr style='mso-yfti-irow:24'>
  <td width=122 valign=top style='width:121.5pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1 style='tab-stops:90.75pt'><span style='font-family:Arial;
  mso-bidi-font-family:"Times New Roman"'><i>gene_collapse_method</i></p>
  </td>
  <td width=75 valign=top style='width:75.1pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>-gcm</p>
  </td>
  <td width=269 valign=top style='width:268.8pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>Method
  for reducing marker-level copy number data to the gene-level copy number data in the gene tables. 
  Markers contained in the gene are used when available, otherwise the flanking 
  marker or markers are used. Allowed values are mean, median, min, max or extreme. 
  The extreme method chooses whichever of min or max is furthest from diploid. 
  (DEFAULT=mean) </p>
  </td>
 </tr>

 <tr style='mso-yfti-irow:25'>
  <td width=122 valign=top style='width:121.5pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1 style='tab-stops:90.75pt'><span style='font-family:Arial;
  mso-bidi-font-family:"Times New Roman"'><i>save_seg_data</i></span><span
  style='font-family:"Times New Roman";mso-bidi-font-family:"Times New Roman"'>
  </p>
  </td>
  <td width=75 valign=top style='width:75.1pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>-saveseg
  </p>
  </td>
  <td width=269 valign=top style='width:268.8pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>Flag
  indicating that the preprocessed segmented data used as input for the GISTIC analysis
  should be saved (in matlab format). Allowed values= {1,0}. (DEFAULT=1, save segmented input
  data) </p>
  </td>
 </tr>

 <tr style='mso-yfti-irow:26'>
  <td width=122 valign=top style='width:121.5pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1 style='tab-stops:90.75pt'><span style='font-family:Arial;
  mso-bidi-font-family:"Times New Roman"'><i>save_data_files</i> </p>
  </td>
  <td width=75 valign=top style='width:75.1pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>-savedata
  </p>
  </td>
  <td width=269 valign=top style='width:268.8pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>Flag
  indicating that native <span style='font-variant:small-caps'>Matlab<sup>TM</sup></span>
  output files should be saved in addition to text data. Allowed values= {1,0}.
  (DEFAULT=1, save <span style='font-variant:small-caps'>Matlab<sup>TM</sup></span>
  files) </p>
  </td>
 </tr>

 <tr style='mso-yfti-irow:27;mso-yfti-lastrow:yes'>
  <td width=122 valign=top style='width:121.5pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1 style='tab-stops:90.75pt'><span style='font-family:Arial;
  mso-bidi-font-family:"Times New Roman"'>use set_verbose_level()</p>
  </td>
  <td width=75 valign=top style='width:75.1pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>-v
  </p>
  </td>
  <td width=269 valign=top style='width:268.8pt;padding:.1in 5.75pt .1in 5.75pt'>
  <p class=msonormal1><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>Integer
  value indicating the level of verbosity to use in the program execution
  log.<span style="mso-spacerun: yes">&nbsp; </span>Suggested values =
  {0,10,20,30}.<span style="mso-spacerun: yes">&nbsp; </span>0 sets no
  verbosity; 30 sets high level of verbosity.<span style="mso-spacerun:
  yes">&nbsp; </span>(DEFAULT=0) </p>
  </td>
 </tr>

</table>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal;mso-bidi-font-weight:
normal'><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></b></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal;mso-bidi-font-weight:
normal'><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></b></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal;mso-bidi-font-weight:
normal'><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></b></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal;mso-bidi-font-weight:
normal'><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>Input
Files</b></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal;mso-bidi-font-weight:
normal'><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></b></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal;mso-bidi-font-weight:
normal'><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></b></p>

<ol style='margin-top:0in' start=1 type=1>
 <li class=MsoNormal style='mso-list:l7 level1 lfo2;tab-stops:list .5in'><b
     style='mso-bidi-font-weight:normal;mso-bidi-font-weight:normal'><span
     style='font-family:Arial'>Segmentation File (-seg)<span style='mso-tab-count:
     4'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>REQUIRED
     </b></li>
</ol>

<p class=MsoNormal style='margin-left:.25in'><b style='mso-bidi-font-weight:
normal;mso-bidi-font-weight:normal'><span style='font-family:Arial;mso-bidi-font-family:
"Times New Roman"'><o:p>&nbsp;</o:p></span></b></p>

<p class=MsoNormal style='margin-left:63.0pt'><span style='font-family:Arial;
mso-bidi-font-family:"Times New Roman"'>The segmentation file contains the
segmented data for all the samples identified by GLAD, CBS, or some other
segmentation algorithm.<span style="mso-spacerun: yes">&nbsp; </span>(See GLAD
file format in the Genepattern file formats documentation.)<span
style="mso-spacerun: yes">&nbsp; </span>It is a six column, tab-delimited file
with an optional first line identifying the columns.<span style="mso-spacerun:
yes">&nbsp; </span>Positions are in base pair units.</p>

<p class=MsoNormal style='margin-left:63.0pt'><span style='font-family:Arial;
mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-left:63.0pt'><span style='font-family:Arial;
mso-bidi-font-family:"Times New Roman"'><span style="mso-spacerun: yes">&nbsp;
</span>The column headers are:</p>

<p class=MsoNormal style='margin-left:63.0pt'><span style='font-family:Arial;
mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-left:1.25in;text-indent:0in;mso-list:l6 level1 lfo3;
tab-stops:list 1.25in'><![if !supportLists]><span style='font-family:Arial;
mso-fareast-font-family:Arial;mso-bidi-font-family:Arial'><span
style='mso-list:Ignore'>(1)<span style='font:7.0pt "Times New Roman"'>&nbsp; </span></span></span><![endif]><i
style='mso-bidi-font-style:normal;mso-bidi-font-style:normal'><span
style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>Sample</span></i><span
style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun:
yes">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span>(sample name)</p>

<p class=MsoNormal style='margin-left:1.25in;text-indent:0in;mso-list:l6 level1 lfo3;
tab-stops:list 1.25in'><![if !supportLists]><span style='font-family:Arial;
mso-fareast-font-family:Arial;mso-bidi-font-family:Arial'><span
style='mso-list:Ignore'>(2)<span style='font:7.0pt "Times New Roman"'>&nbsp; </span></span></span><![endif]><i
style='mso-bidi-font-style:normal;mso-bidi-font-style:normal'><span
style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>Chromosome</span></i><span
style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">&nbsp; </span>(chromosome number)</p>

<p class=MsoNormal style='margin-left:1.25in;text-indent:0in;mso-list:l6 level1 lfo3;
tab-stops:list 1.25in'><![if !supportLists]><span style='font-family:Arial;
mso-fareast-font-family:Arial;mso-bidi-font-family:Arial'><span
style='mso-list:Ignore'>(3)<span style='font:7.0pt "Times New Roman"'>&nbsp; </span></span></span><![endif]><i
style='mso-bidi-font-style:normal;mso-bidi-font-style:normal'><span
style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>Start Position</span></i><span
style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">&nbsp; </span>(segment start position, in bases)</p>

<p class=MsoNormal style='margin-left:1.25in;text-indent:0in;mso-list:l6 level1 lfo3;
tab-stops:list 1.25in'><![if !supportLists]><span style='font-family:Arial;
mso-fareast-font-family:Arial;mso-bidi-font-family:Arial'><span
style='mso-list:Ignore'>(4)<span style='font:7.0pt "Times New Roman"'>&nbsp; </span></span></span><![endif]><i
style='mso-bidi-font-style:normal;mso-bidi-font-style:normal'><span
style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>End Position</span></i><span
style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">&nbsp;&nbsp; </span>(segment end position, in bases)</p>

<p class=MsoNormal style='margin-left:1.25in;text-indent:0in;mso-list:l6 level1 lfo3;
tab-stops:list 1.25in'><![if !supportLists]><span style='font-family:Arial;
mso-fareast-font-family:Arial;mso-bidi-font-family:Arial'><span
style='mso-list:Ignore'>(5)<span style='font:7.0pt "Times New Roman"'>&nbsp; </span></span></span><![endif]><i
style='mso-bidi-font-style:normal;mso-bidi-font-style:normal'><span
style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>Num markers</span></i><span
style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>(number of
markers in segment)</p>

<p class=MsoNormal style='margin-left:1.25in;text-indent:0in;mso-list:l6 level1 lfo3;
tab-stops:list 1.25in'><![if !supportLists]><span style='font-family:Arial;
mso-fareast-font-family:Arial;mso-bidi-font-family:Arial'><span
style='mso-list:Ignore'>(6)<span style='font:7.0pt "Times New Roman"'>&nbsp; </span></span></span><![endif]><i
style='mso-bidi-font-style:normal;mso-bidi-font-style:normal'><span
style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>Seg.CN</span></i><span
style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><span
style="mso-spacerun: yes">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>(log2()
-1 of copy number)</p>

<p class=MsoNormal><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-left:.5in;text-indent:.5in'><span
style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><a
href="examplefiles/segmentationfile.txt">Example Segmentation File</a></p>

<p class=MsoNormal><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<ol style='margin-top:0in' start=2 type=1>
 <li class=MsoNormal style='mso-list:l7 level1 lfo2;tab-stops:list .5in'><b
     style='mso-bidi-font-weight:normal;mso-bidi-font-weight:normal'><span
     style='font-family:Arial'>Markers File (-mk)<span style='mso-tab-count:
     5'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>OPTIONAL
     </b></li>
</ol>

<p class=MsoNormal style='margin-left:.25in'><b style='mso-bidi-font-weight:
normal;mso-bidi-font-weight:normal'><span style='font-family:Arial;mso-bidi-font-family:
"Times New Roman"'><o:p>&nbsp;</o:p></span></b></p>

<p class=MsoNormal style='margin-left:63.0pt'><span style='font-family:Arial;
mso-bidi-font-family:"Times New Roman"'>The markers file identifies the marker
positions used in the original dataset (before
segmentation) for array or capture experiments. As of GISTIC release 2.0.23,
the markers file is an optional input - if omitted, pseudo markers are generated 
as uniformly as possible using the <em>maxspace</em> input parameter.
<span style="mso-spacerun: yes">&nbsp; </span></p>

<p class=MsoNormal style='margin-left:63.0pt'><span style='font-family:Arial;
mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-left:63.0pt'><span style='font-family:Arial;
mso-bidi-font-family:"Times New Roman"'>
The markers file is a three
column, tab-delimited file with an optional header.<span style="mso-spacerun:
yes">&nbsp; </span>The column headers are:</p>

<p class=MsoNormal style='margin-left:63.0pt'><span style='font-family:Arial;
mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-left:63.0pt;text-indent:27.0pt;mso-list:l3 level1 lfo4;
tab-stops:list 37.5pt'><![if !supportLists]><i style='mso-bidi-font-style:normal;
mso-bidi-font-style:normal'><span style='font-family:Arial;mso-fareast-font-family:
Arial;mso-bidi-font-family:Arial'><span style='mso-list:Ignore'>(1)<span
style='font:7.0pt "Times New Roman"'>&nbsp; </span></span></span></i><![endif]><i
style='mso-bidi-font-style:normal'><span style='font-family:Arial;mso-bidi-font-family:
"Times New Roman"'>Marker Name</i></p>

<p class=MsoNormal style='margin-left:63.0pt;text-indent:27.0pt;mso-list:l3 level1 lfo4;
tab-stops:list 37.5pt'><![if !supportLists]><i style='mso-bidi-font-style:normal;
mso-bidi-font-style:normal'><span style='font-family:Arial;mso-fareast-font-family:
Arial;mso-bidi-font-family:Arial'><span style='mso-list:Ignore'>(2)<span
style='font:7.0pt "Times New Roman"'>&nbsp; </span></span></span></i><![endif]><i
style='mso-bidi-font-style:normal'><span style='font-family:Arial;mso-bidi-font-family:
"Times New Roman"'>Chromosome</i></p>

<p class=MsoNormal style='margin-left:63.0pt;text-indent:27.0pt;mso-list:l3 level1 lfo4;
tab-stops:list 37.5pt'><![if !supportLists]><i style='mso-bidi-font-style:normal;
mso-bidi-font-style:normal'><span style='font-family:Arial;mso-fareast-font-family:
Arial;mso-bidi-font-family:Arial'><span style='mso-list:Ignore'>(3)<span
style='font:7.0pt "Times New Roman"'>&nbsp; </span></span></span></i><![endif]><i
style='mso-bidi-font-style:normal'><span style='font-family:Arial;mso-bidi-font-family:
"Times New Roman"'>Marker Position<span style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span></span></i><span
style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>(in bases)<i
style='mso-bidi-font-style:normal;mso-bidi-font-style:normal'><o:p></o:p></i></span></p>

<p class=MsoNormal style='margin-left:.25in'><b style='mso-bidi-font-weight:
normal;mso-bidi-font-weight:normal'><span style='font-family:Arial;mso-bidi-font-family:
"Times New Roman"'><o:p>&nbsp;</o:p></span></b></p>

<p class=MsoNormal style='margin-left:63.0pt'><span style='font-family:Arial;
mso-bidi-font-family:"Times New Roman"'><a href="examplefiles/markersfile.txt">Example
Markers File</a></p>

<p class=MsoNormal style='margin-left:63.0pt'><b style='mso-bidi-font-weight:
normal;mso-bidi-font-weight:normal'><span style='font-family:Arial;mso-bidi-font-family:
"Times New Roman"'><o:p>&nbsp;</o:p></span></b></p>

<ol style='margin-top:0in' start=3 type=1>
 <li class=MsoNormal style='mso-list:l7 level1 lfo2;tab-stops:list .5in'><b
     style='mso-bidi-font-weight:normal;mso-bidi-font-weight:normal'><span
     style='font-family:Arial'>Reference Genome File (-refgene)<span
     style='mso-tab-count:4'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>
     REQUIRED </b></li>
</ol>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal;mso-bidi-font-weight:
normal'><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.0in'><span style='font-family:Arial;
mso-bidi-font-family:"Times New Roman"'>The reference genome file contains
information about the location of genes and cytobands on a given build of the
genome. Reference genome files are created in <span style='font-variant:small-caps'>Matlab<sup>TM</sup></span>
and are not viewable with a text editor. The GISTIC 2.0 release has four
reference genomes located in the <i style='mso-bidi-font-style:normal;
mso-bidi-font-style:normal'>refgenefiles</i> directory: <i style='mso-bidi-font-style:
normal;mso-bidi-font-style:normal'>hg16.mat</i>, <i style='mso-bidi-font-style:
normal;mso-bidi-font-style:normal'>hg17.mat</i>, <i style='mso-bidi-font-style:
normal;mso-bidi-font-style:normal'>hg18.mat</i>, and <i style='mso-bidi-font-style:
normal;mso-bidi-font-style:normal'>hg19.mat</i>.<span style="mso-spacerun:
yes">&nbsp; </span></p>

<p class=MsoNormal><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<ol style='margin-top:0in' start=4 type=1>
 <li class=MsoNormal style='mso-list:l7 level1 lfo2;tab-stops:list .5in'><b
     style='mso-bidi-font-weight:normal;mso-bidi-font-weight:normal'><span
     style='font-family:Arial'>Array List File (-alf)<span style='mso-tab-count:
     4'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>OPTIONAL</b></li>
</ol>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal;mso-bidi-font-weight:
normal'><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.0in'><span style='font-family:Arial;
mso-bidi-font-family:"Times New Roman"'>The array list file is an optional file
identifying the subset of samples to be used in the analysis.<span
style="mso-spacerun: yes">&nbsp; </span>It is a one column file with an
optional header (<i style='mso-bidi-font-style:normal;mso-bidi-font-style:normal'>array</i>).<span
style="mso-spacerun: yes">&nbsp; </span>The sample identifiers listed in the
array list file must match the sample names given in the segmentation
file.<span style="mso-spacerun: yes">&nbsp; </span></p>

<p class=MsoNormal><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><span
style='mso-tab-count:2'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span><a
href="examplefiles/arraylistfile.txt">Example Array List File</a></p>

<p class=MsoNormal><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<ol style='margin-top:0in' start=5 type=1>
 <li class=MsoNormal style='mso-list:l7 level1 lfo2;tab-stops:list .5in'><b
     style='mso-bidi-font-weight:normal;mso-bidi-font-weight:normal'><span
     style='font-family:Arial'>CNV File (-cnv)<span style='mso-tab-count:5'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>OPTIONAL
     </b></li>
</ol>

<p class=MsoNormal style='margin-left:.25in'><span style='font-family:Arial;
mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-left:1.0in'><span style='font-family:Arial;
mso-bidi-font-family:"Times New Roman"'>There are two options for the file specifying germ line CNVs
to be excluded from the analysis.<span style="mso-spacerun: yes">&nbsp; </span>The first option allows CNVs
to be identified by marker name and is platform-specific.<span style="mso-spacerun: yes">&nbsp;
</span>The second option allows the CNVs to be identified by genomic location, which is platform independent but genome-build dependent.</p>

<p class=MsoNormal style='margin-left:1.0in'><span style='font-family:Arial;
mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-left:1.0in'><u><span style='font-family:Arial;
mso-bidi-font-family:"Times New Roman"'>Option #1</span></u><span
style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>:<span
style="mso-spacerun: yes">&nbsp; </span>A two column, tab-delimited file with
an optional header row.<span style="mso-spacerun: yes">&nbsp; </span>The marker
names given in this file must match the marker names given in the markers
file.<span style="mso-spacerun: yes">&nbsp; </span>The CNV identifiers are for
user use and can be arbitrary.<span style="mso-spacerun: yes">&nbsp; </span>The
column headers are: </p>

<p class=MsoNormal style='margin-left:1.25in;text-indent:0in;mso-list:l1 level1 lfo5;
tab-stops:list 34.5pt'><![if !supportLists]><span style='font-family:Arial;
mso-fareast-font-family:Arial;mso-bidi-font-family:Arial'><span
style='mso-list:Ignore'>(1)<span style='font:7.0pt "Times New Roman"'>&nbsp; </span></span></span><![endif]><i
style='mso-bidi-font-style:normal;mso-bidi-font-style:normal'><span
style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>Marker Name </i></p>

<p class=MsoNormal style='margin-left:1.25in;text-indent:0in;mso-list:l1 level1 lfo5;
tab-stops:list 34.5pt'><![if !supportLists]><span style='font-family:Arial;
mso-fareast-font-family:Arial;mso-bidi-font-family:Arial'><span
style='mso-list:Ignore'>(2)<span style='font:7.0pt "Times New Roman"'>&nbsp; </span></span></span><![endif]><i
style='mso-bidi-font-style:normal;mso-bidi-font-style:normal'><span
style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>CNV Identifier</i></p>

<p class=MsoNormal style='margin-left:1.0in'><i style='mso-bidi-font-style:
normal;mso-bidi-font-style:normal'><span style='font-family:Arial;mso-bidi-font-family:
"Times New Roman"'><o:p>&nbsp;</o:p></span></i></p>

<p class=MsoNormal style='margin-left:1.0in'><u><span style='font-family:Arial;
mso-bidi-font-family:"Times New Roman"'>Option #2</span></u><span
style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>:<span
style="mso-spacerun: yes">&nbsp; </span>A 6 column, tab-delimited file with an
optional header row.<span style="mso-spacerun: yes">&nbsp; </span>The ‘CNV
Identifier’ is for user use and can be arbitrary.<span style="mso-spacerun:
yes">&nbsp; </span>‘Narrow Region Start’ and ‘Narrow Region End’ are also not
used.<span style="mso-spacerun: yes">&nbsp; </span>The column headers are:</p>

<p class=MsoNormal style='margin-left:1.0in;text-indent:.25in;mso-list:l4 level1 lfo6;
tab-stops:list 37.5pt'><![if !supportLists]><span style='font-family:Arial;
mso-fareast-font-family:Arial;mso-bidi-font-family:Arial'><span
style='mso-list:Ignore'>(1)<span style='font:7.0pt "Times New Roman"'>&nbsp; </span></span></span><![endif]><i
style='mso-bidi-font-style:normal;mso-bidi-font-style:normal'><span
style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>CNV Identifier</i></p>

<p class=MsoNormal style='margin-left:1.0in;text-indent:.25in;mso-list:l4 level1 lfo6;
tab-stops:list 37.5pt'><![if !supportLists]><span style='font-family:Arial;
mso-fareast-font-family:Arial;mso-bidi-font-family:Arial'><span
style='mso-list:Ignore'>(2)<span style='font:7.0pt "Times New Roman"'>&nbsp; </span></span></span><![endif]><i
style='mso-bidi-font-style:normal;mso-bidi-font-style:normal'><span
style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>Chromosome</span></i><span
style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'></p>

<p class=MsoNormal style='margin-left:1.0in;text-indent:.25in;mso-list:l4 level1 lfo6;
tab-stops:list 37.5pt'><![if !supportLists]><span style='font-family:Arial;
mso-fareast-font-family:Arial;mso-bidi-font-family:Arial'><span
style='mso-list:Ignore'>(3)<span style='font:7.0pt "Times New Roman"'>&nbsp; </span></span></span><![endif]><i
style='mso-bidi-font-style:normal;mso-bidi-font-style:normal'><span
style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>Narrow Region
Start</span></i><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'></p>

<p class=MsoNormal style='margin-left:1.0in;text-indent:.25in;mso-list:l4 level1 lfo6;
tab-stops:list 37.5pt'><![if !supportLists]><span style='font-family:Arial;
mso-fareast-font-family:Arial;mso-bidi-font-family:Arial'><span
style='mso-list:Ignore'>(4)<span style='font:7.0pt "Times New Roman"'>&nbsp; </span></span></span><![endif]><i
style='mso-bidi-font-style:normal;mso-bidi-font-style:normal'><span
style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>Narrow Region
End</span></i><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'></p>

<p class=MsoNormal style='margin-left:1.0in;text-indent:.25in;mso-list:l4 level1 lfo6;
tab-stops:list 37.5pt'><![if !supportLists]><span style='font-family:Arial;
mso-fareast-font-family:Arial;mso-bidi-font-family:Arial'><span
style='mso-list:Ignore'>(5)<span style='font:7.0pt "Times New Roman"'>&nbsp; </span></span></span><![endif]><i
style='mso-bidi-font-style:normal;mso-bidi-font-style:normal'><span
style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>Wide Region
Start</span></i><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'></p>

<p class=MsoNormal style='margin-left:1.0in;text-indent:.25in;mso-list:l4 level1 lfo6;
tab-stops:list 37.5pt'><![if !supportLists]><span style='font-family:Arial;
mso-fareast-font-family:Arial;mso-bidi-font-family:Arial'><span
style='mso-list:Ignore'>(6)<span style='font:7.0pt "Times New Roman"'>&nbsp; </span></span></span><![endif]><i
style='mso-bidi-font-style:normal;mso-bidi-font-style:normal'><span
style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>Wide Region
End</span></i><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal;mso-bidi-font-weight:
normal'><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></b></p>

<p class=MsoNormal style='margin-left:.5in'><span style='font-family:Arial;
mso-bidi-font-family:"Times New Roman"'><a href="examplefiles/cnvfile.txt">Example
CNV File</a></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal;mso-bidi-font-weight:
normal'><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></b></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal;mso-bidi-font-weight:
normal'><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></b></p>

<b style='mso-bidi-font-weight:normal;mso-bidi-font-weight:normal'><span
style='font-size:12.0pt;font-family:Arial;mso-fareast-font-family:"Times New Roman";
mso-bidi-font-family:"Times New Roman";mso-bidi-theme-font:minor-bidi;
mso-ansi-language:EN-US;mso-fareast-language:EN-US'><br clear=ALL
style='page-break-before:always'>
</span></b>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal;mso-bidi-font-weight:
normal'><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>Output
Files</b></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal;mso-bidi-font-weight:
normal'><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></b></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal;mso-bidi-font-weight:
normal'><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></b></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal;mso-bidi-font-weight:
normal'><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>1.
All Lesions File </span></b><span style='font-family:Arial;mso-bidi-font-family:
"Times New Roman"'>(all_lesions.conf_<i style='mso-bidi-font-style:normal;
mso-bidi-font-style:normal'>XX</i>.txt, where <i style='mso-bidi-font-style:
normal;mso-bidi-font-style:normal'>XX</i> is the confidence level)</p>

<p class=MsoNormal><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>The
all lesions file summarizes the results from the GISTIC run.<span
style="mso-spacerun: yes">&nbsp; </span>It contains data about the significant
regions of amplification and deletion as well as which samples are amplified or
deleted in each of these regions.<span style="mso-spacerun: yes">&nbsp;
</span>The identified regions are listed down the first column, and the samples
are listed across the first row, starting in column 10.<span
style="mso-spacerun: yes">&nbsp; </span></p>

<p class=MsoNormal><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><i style='mso-bidi-font-style:normal;mso-bidi-font-style:
normal'><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>Region
Data</i></p>

<p class=MsoNormal><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>Columns
1-9 present the data about the significant regions as follows:</p>

<p class=MsoNormal style='margin-left:27.75pt;text-indent:-.25in;mso-list:l5 level1 lfo7;
tab-stops:list 27.75pt'><![if !supportLists]><span style='font-family:Arial;
mso-fareast-font-family:Arial;mso-bidi-font-family:Arial'><span
style='mso-list:Ignore'>(1)<span style='font:7.0pt "Times New Roman"'>&nbsp; </span></span></span><![endif]><i
style='mso-bidi-font-style:normal;mso-bidi-font-style:normal'><span
style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>Unique Name: </span></i><span
style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>A name
assigned to identify the region</p>

<p class=MsoNormal style='margin-left:27.75pt;text-indent:-.25in;mso-list:l5 level1 lfo7;
tab-stops:list 27.75pt'><![if !supportLists]><span style='font-family:Arial;
mso-fareast-font-family:Arial;mso-bidi-font-family:Arial'><span
style='mso-list:Ignore'>(2)<span style='font:7.0pt "Times New Roman"'>&nbsp; </span></span></span><![endif]><i
style='mso-bidi-font-style:normal;mso-bidi-font-style:normal'><span
style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>Descriptor: </span></i><span
style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>The genomic
descriptor of that region.</p>

<p class=MsoNormal style='margin-left:27.75pt;text-indent:-.25in;mso-list:l5 level1 lfo7;
tab-stops:list 27.75pt'><![if !supportLists]><span style='font-family:Arial;
mso-fareast-font-family:Arial;mso-bidi-font-family:Arial'><span
style='mso-list:Ignore'>(3)<span style='font:7.0pt "Times New Roman"'>&nbsp; </span></span></span><![endif]><i
style='mso-bidi-font-style:normal;mso-bidi-font-style:normal'><span
style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>Wide Peak
Limits: </span></i><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>The
“wide peak” boundaries most likely to contain the targeted genes.<span
style="mso-spacerun: yes">&nbsp; </span>These are listed in genomic coordinates
and marker (or probe) indices.</p>

<p class=MsoNormal style='margin-left:27.75pt;text-indent:-.25in;mso-list:l5 level1 lfo7;
tab-stops:list 27.75pt'><![if !supportLists]><span style='font-family:Arial;
mso-fareast-font-family:Arial;mso-bidi-font-family:Arial'><span
style='mso-list:Ignore'>(4)<span style='font:7.0pt "Times New Roman"'>&nbsp; </span></span></span><![endif]><i
style='mso-bidi-font-style:normal;mso-bidi-font-style:normal'><span
style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>Peak Limits: </span></i><span
style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>The boundaries
of the region of maximal amplification or deletion.</p>

<p class=MsoNormal style='margin-left:27.75pt;text-indent:-.25in;mso-list:l5 level1 lfo7;
tab-stops:list 27.75pt'><![if !supportLists]><span style='font-family:Arial;
mso-fareast-font-family:Arial;mso-bidi-font-family:Arial'><span
style='mso-list:Ignore'>(5)<span style='font:7.0pt "Times New Roman"'>&nbsp; </span></span></span><![endif]><i
style='mso-bidi-font-style:normal;mso-bidi-font-style:normal'><span
style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>Region Limits:
</span></i><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>The
boundaries of the entire significant region of amplification or deletion.</p>

<p class=MsoNormal style='margin-left:27.75pt;text-indent:-.25in;mso-list:l5 level1 lfo7;
tab-stops:list 27.75pt'><![if !supportLists]><span style='font-family:Arial;
mso-fareast-font-family:Arial;mso-bidi-font-family:Arial'><span
style='mso-list:Ignore'>(6)<span style='font:7.0pt "Times New Roman"'>&nbsp; </span></span></span><![endif]><i
style='mso-bidi-font-style:normal;mso-bidi-font-style:normal'><span
style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>q-values: </span></i><span
style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>The q-value of
the peak region.</p>

<p class=MsoNormal style='margin-left:27.75pt;text-indent:-.25in;mso-list:l5 level1 lfo7;
tab-stops:list 27.75pt'><![if !supportLists]><span style='font-family:Arial;
mso-fareast-font-family:Arial;mso-bidi-font-family:Arial'><span
style='mso-list:Ignore'>(7)<span style='font:7.0pt "Times New Roman"'>&nbsp; </span></span></span><![endif]><i
style='mso-bidi-font-style:normal;mso-bidi-font-style:normal'><span
style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>Residual
q-values: </span></i><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>The
q-value of the peak region after removing (“peeling off”) amplifications or
deletions that overlap other, more significant peak regions in the same
chromosome.</p>

<p class=MsoNormal style='margin-left:27.75pt;text-indent:-.25in;mso-list:l5 level1 lfo7;
tab-stops:list 27.75pt'><![if !supportLists]><span style='font-family:Arial;
mso-fareast-font-family:Arial;mso-bidi-font-family:Arial'><span
style='mso-list:Ignore'>(8)<span style='font:7.0pt "Times New Roman"'>&nbsp; </span></span></span><![endif]><i
style='mso-bidi-font-style:normal;mso-bidi-font-style:normal'><span
style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>Broad or
Focal:</span></i><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>
Identifies whether the region reaches significance due primarily to broad
events (called “broad”), focal events (called “focal”), or independently
significant broad and focal events (called “both”).</p>

<p class=MsoNormal style='margin-left:27.75pt;text-indent:-.25in;mso-list:l5 level1 lfo7;
tab-stops:list 27.75pt'><![if !supportLists]><span style='font-family:Arial;
mso-fareast-font-family:Arial;mso-bidi-font-family:Arial'><span
style='mso-list:Ignore'>(9)<span style='font:7.0pt "Times New Roman"'>&nbsp; </span></span></span><![endif]><i
style='mso-bidi-font-style:normal;mso-bidi-font-style:normal'><span
style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>Amplitude
Threshold:</span></i><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>
Key giving the meaning of values in the subsequent columns associated with each
sample.</p>

<p class=MsoNormal><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><i style='mso-bidi-font-style:normal;mso-bidi-font-style:
normal'><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>Sample
Data</span></i><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>
</p>

<p class=MsoNormal><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>Each
of the analyzed samples is represented in one of the columns following the
lesion data (columns 10 through end).<span style="mso-spacerun: yes">&nbsp;
</span>The data contained in these columns varies slightly by section of the
file.<span style="mso-spacerun: yes">&nbsp; </span></p>

<p class=MsoNormal><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>The
first section can be identified by the key given in column 9 &#8211; it starts
in row 2 and continues until the row that reads “Actual Copy Change
Given.”<span style="mso-spacerun: yes">&nbsp; </span>This section contains
summarized data for each sample.<span style="mso-spacerun: yes">&nbsp; </span>A
‘0’ indicates that the copy number of the sample was not amplified or deleted
beyond the threshold amount in that peak region.<span style="mso-spacerun:
yes">&nbsp; </span>A ‘1’ indicates that the sample had low-level copy number
aberrations (exceeding the low threshold indicated in column 9), and a ‘2’
indicates that the sample had high-level copy number aberrations (exceeding the
high threshold indicated in column 9).<i style='mso-bidi-font-style:normal;
mso-bidi-font-style:normal'><o:p></o:p></i></span></p>

<p class=MsoNormal><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>The
second section can be identified the rows in which column 9 reads “Actual Copy
Change Given.”<span style="mso-spacerun: yes">&nbsp; </span>The second section
exactly reproduces the first section, except that here the actual changes in
copy number are provided rather than zeroes, ones, and twos.</p>

<p class=MsoNormal><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>The
final section is similar to the first section, except that here only broad
events are included.<span style="mso-spacerun: yes">&nbsp; </span>A 1 in the
samples columns (columns 10+) indicates that the median copy number of the
sample across the entire significant region exceeded the threshold given in
column 9.<span style="mso-spacerun: yes">&nbsp; </span>That is, it indicates
whether the sample had a geographically extended event, rather than a focal
amplification or deletion covering little more than the peak region.</p>

<p class=MsoNormal><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><!--[if mso & !supportInlineShapes & supportFields]><span
style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><span
style='mso-element:field-begin;mso-field-lock:yes'></span><span
style="mso-spacerun: yes">&nbsp;</span>SHAPE <span style="mso-spacerun:
yes">&nbsp;</span>\* MERGEFORMAT <span style='mso-element:field-separator'></span></span><![endif]--><span
style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><!--[if gte vml 1]><v:group
 id="_x0000_s1091" style='width:544.8pt;height:263pt;
 mso-position-horizontal-relative:char;mso-position-vertical-relative:line'
 coordorigin="-108,2880" coordsize="11760,5250">
 <v:group id="_x0000_s1092" style='position:absolute;left:7632;top:2880;
  width:3900;height:540' coordorigin="7632,2880" coordsize="3900,540">
  <v:shapetype id="_x0000_t87" coordsize="21600,21600" o:spt="87" adj="1800,10800"
   path="m21600,0qx10800@0l10800@2qy0@11,10800@3l10800@1qy21600,21600e"
   filled="f">
   <v:formulas>
    <v:f eqn="val #0"/>
    <v:f eqn="sum 21600 0 #0"/>
    <v:f eqn="sum #1 0 #0"/>
    <v:f eqn="sum #1 #0 0"/>
    <v:f eqn="prod #0 9598 32768"/>
    <v:f eqn="sum 21600 0 @4"/>
    <v:f eqn="sum 21600 0 #1"/>
    <v:f eqn="min #1 @6"/>
    <v:f eqn="prod @7 1 2"/>
    <v:f eqn="prod #0 2 1"/>
    <v:f eqn="sum 21600 0 @9"/>
    <v:f eqn="val #1"/>
   </v:formulas>
   <v:path arrowok="t" o:connecttype="custom" o:connectlocs="21600,0;0,10800;21600,21600"
    textboxrect="13963,@4,21600,@5"/>
   <v:handles>
    <v:h position="center,#0" yrange="0,@8"/>
    <v:h position="topLeft,#1" yrange="@9,@10"/>
   </v:handles>
  </v:shapetype><v:shape id="_x0000_s1093" type="#_x0000_t87" style='position:absolute;
   left:9492;top:1380;width:180;height:3900;rotation:-90;flip:y' strokecolor="#396"
   strokeweight="2pt"/>
  <v:shapetype id="_x0000_t202" coordsize="21600,21600" o:spt="202" path="m0,0l0,21600,21600,21600,21600,0xe">
   <v:stroke joinstyle="miter"/>
   <v:path gradientshapeok="t" o:connecttype="rect"/>
  </v:shapetype><v:shape id="_x0000_s1094" type="#_x0000_t202" style='position:absolute;
   left:8532;top:2880;width:1980;height:540' filled="f" stroked="f"
   strokecolor="#396" strokeweight="2pt">
   <v:textbox style='mso-next-textbox:#_x0000_s1094'>
    <![if !mso]>
    <table cellpadding=0 cellspacing=0 width="100%">
     <tr>
      <td><![endif]>
      <div>
      <p class=MsoNormal><b style='mso-bidi-font-weight:normal;mso-bidi-font-weight:
      normal'><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman";
      color:#339966'>Sample Data</b></p>
      </div>
      <![if !mso]></td>
     </tr>
    </table>
    <![endif]></v:textbox>
  </v:shape></v:group><v:shapetype id="_x0000_t75" coordsize="21600,21600"
  o:spt="75" o:preferrelative="t" path="m@4@5l@4@11@9@11@9@5xe" filled="f"
  stroked="f">
  <v:stroke joinstyle="miter"/>
  <v:formulas>
   <v:f eqn="if lineDrawn pixelLineWidth 0"/>
   <v:f eqn="sum @0 1 0"/>
   <v:f eqn="sum 0 0 @1"/>
   <v:f eqn="prod @2 1 2"/>
   <v:f eqn="prod @3 21600 pixelWidth"/>
   <v:f eqn="prod @3 21600 pixelHeight"/>
   <v:f eqn="sum @0 0 1"/>
   <v:f eqn="prod @6 1 2"/>
   <v:f eqn="prod @7 21600 pixelWidth"/>
   <v:f eqn="sum @8 21600 0"/>
   <v:f eqn="prod @7 21600 pixelHeight"/>
   <v:f eqn="sum @10 21600 0"/>
  </v:formulas>
  <v:path o:extrusionok="f" gradientshapeok="t" o:connecttype="rect"/>
  <o:lock v:ext="edit" aspectratio="t"/>
 </v:shapetype><v:shape id="_x0000_s1095" type="#_x0000_t75" style='position:absolute;
  left:1332;top:3420;width:10320;height:4710'>
  <v:imagedata src="" o:title="" croptop="12227f" cropbottom="22487f"
   cropleft="1081f" cropright="34947f"/>
 </v:shape><v:group id="_x0000_s1096" style='position:absolute;left:1332;top:2880;
  width:6121;height:540' coordorigin="2232,5220" coordsize="6841,540">
  <v:shape id="_x0000_s1097" type="#_x0000_t87" style='position:absolute;
   left:5563;top:2249;width:180;height:6841;rotation:-90;flip:y' strokecolor="#396"
   strokeweight="2pt"/>
  <v:shape id="_x0000_s1098" type="#_x0000_t202" style='position:absolute;
   left:4750;top:5220;width:2342;height:360' filled="f" stroked="f">
   <v:textbox style='mso-next-textbox:#_x0000_s1098'>
    <![if !mso]>
    <table cellpadding=0 cellspacing=0 width="100%">
     <tr>
      <td><![endif]>
      <div>
      <p class=MsoNormal><b style='mso-bidi-font-weight:normal;mso-bidi-font-weight:
      normal'><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman";
      color:#339966'>Lesion Data</b></p>
      </div>
      <![if !mso]></td>
     </tr>
    </table>
    <![endif]></v:textbox>
  </v:shape></v:group><v:group id="_x0000_s1099" style='position:absolute;
  left:-108;top:3600;width:1440;height:1800' coordorigin="1332,9720"
  coordsize="1440,1800">
  <v:shape id="_x0000_s1100" type="#_x0000_t87" style='position:absolute;
   left:2592;top:9720;width:180;height:1800' strokecolor="#396" strokeweight="2pt"/>
  <v:shape id="_x0000_s1101" type="#_x0000_t202" style='position:absolute;
   left:1332;top:10274;width:1440;height:886' filled="f" stroked="f">
   <v:textbox style='mso-next-textbox:#_x0000_s1101'>
    <![if !mso]>
    <table cellpadding=0 cellspacing=0 width="100%">
     <tr>
      <td><![endif]>
      <div>
      <p class=MsoNormal align=center style='text-align:center'><span
      style='font-family:"Arial Black";mso-bidi-font-family:Arial;color:#339966'>Section
      1</p>
      </div>
      <![if !mso]></td>
     </tr>
    </table>
    <![endif]></v:textbox>
  </v:shape></v:group><v:group id="_x0000_s1102" style='position:absolute;
  left:-108;top:5400;width:1440;height:1800' coordorigin="1332,9720"
  coordsize="1440,1800">
  <v:shape id="_x0000_s1103" type="#_x0000_t87" style='position:absolute;
   left:2592;top:9720;width:180;height:1800' strokecolor="#396" strokeweight="2pt"/>
  <v:shape id="_x0000_s1104" type="#_x0000_t202" style='position:absolute;
   left:1332;top:10274;width:1440;height:886' filled="f" stroked="f">
   <v:textbox style='mso-next-textbox:#_x0000_s1104'>
    <![if !mso]>
    <table cellpadding=0 cellspacing=0 width="100%">
     <tr>
      <td><![endif]>
      <div>
      <p class=MsoNormal align=center style='text-align:center'><span
      style='font-family:"Arial Black";mso-bidi-font-family:Arial;color:#339966'>Section
      2</p>
      </div>
      <![if !mso]></td>
     </tr>
    </table>
    <![endif]></v:textbox>
  </v:shape></v:group><v:group id="_x0000_s1105" style='position:absolute;
  left:-108;top:7200;width:1440;height:900' coordorigin="1332,8280"
  coordsize="1440,900">
  <v:shape id="_x0000_s1106" type="#_x0000_t87" style='position:absolute;
   left:2592;top:8460;width:180;height:720' strokecolor="#396" strokeweight="2pt"/>
  <v:shape id="_x0000_s1107" type="#_x0000_t202" style='position:absolute;
   left:1332;top:8280;width:1440;height:886' filled="f" stroked="f">
   <v:textbox style='mso-next-textbox:#_x0000_s1107'>
    <![if !mso]>
    <table cellpadding=0 cellspacing=0 width="100%">
     <tr>
      <td><![endif]>
      <div>
      <p class=MsoNormal align=center style='text-align:center'><span
      style='font-family:"Arial Black";mso-bidi-font-family:Arial;color:#339966'>Section
      3</p>
      </div>
      <![if !mso]></td>
     </tr>
    </table>
    <![endif]></v:textbox>
  </v:shape></v:group><w:wrap type="none"/>
 <w:anchorlock/>
</v:group><![endif]--></span><!--[if mso & !supportInlineShapes & supportFields]><span
style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><v:shape id="_x0000_i1025"
 type="#_x0000_t75" style='width:544.8pt;height:263pt'>
 <v:imagedata croptop="-65520f" cropbottom="65520f"/>
</v:shape><span style='mso-element:field-end'></span></span><![endif]--><span
style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal;mso-bidi-font-weight:
normal'><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></b></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal;mso-bidi-font-weight:
normal'><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></b></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal;mso-bidi-font-weight:
normal'><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>2.
Amplification Genes File </span></b><span style='font-family:Arial;mso-bidi-font-family:
"Times New Roman"'>(amp_genes.conf_<i style='mso-bidi-font-style:normal;
mso-bidi-font-style:normal'>XX</i>.txt, where <i style='mso-bidi-font-style:
normal;mso-bidi-font-style:normal'>XX</i> is the confidence level)</p>

<p class=MsoNormal><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>The
amp genes file contains one column for each amplification peak identified in
the GISTIC analysis.<span style="mso-spacerun: yes">&nbsp; </span>The first
four rows are:</p>

<p class=MsoNormal style='margin-left:27.75pt;text-indent:-.25in;mso-list:l2 level1 lfo8;
tab-stops:list 27.75pt'><![if !supportLists]><span style='font-family:Arial;
mso-fareast-font-family:Arial;mso-bidi-font-family:Arial'><span
style='mso-list:Ignore'>(1)<span style='font:7.0pt "Times New Roman"'>&nbsp; </span></span></span><![endif]><i
style='mso-bidi-font-style:normal;mso-bidi-font-style:normal'><span
style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>cytoband</span></i><span
style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'></p>

<p class=MsoNormal style='margin-left:27.75pt;text-indent:-.25in;mso-list:l2 level1 lfo8;
tab-stops:list 27.75pt'><![if !supportLists]><span style='font-family:Arial;
mso-fareast-font-family:Arial;mso-bidi-font-family:Arial'><span
style='mso-list:Ignore'>(2)<span style='font:7.0pt "Times New Roman"'>&nbsp; </span></span></span><![endif]><i
style='mso-bidi-font-style:normal;mso-bidi-font-style:normal'><span
style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>q-value</span></i><span
style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'></p>

<p class=MsoNormal style='margin-left:27.75pt;text-indent:-.25in;mso-list:l2 level1 lfo8;
tab-stops:list 27.75pt'><![if !supportLists]><span style='font-family:Arial;
mso-fareast-font-family:Arial;mso-bidi-font-family:Arial'><span
style='mso-list:Ignore'>(3)<span style='font:7.0pt "Times New Roman"'>&nbsp; </span></span></span><![endif]><i
style='mso-bidi-font-style:normal;mso-bidi-font-style:normal'><span
style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>residual
q-value</span></i><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'></p>

<p class=MsoNormal style='margin-left:27.75pt;text-indent:-.25in;mso-list:l2 level1 lfo8;
tab-stops:list 27.75pt'><![if !supportLists]><span style='font-family:Arial;
mso-fareast-font-family:Arial;mso-bidi-font-family:Arial'><span
style='mso-list:Ignore'>(4)<span style='font:7.0pt "Times New Roman"'>&nbsp; </span></span></span><![endif]><i
style='mso-bidi-font-style:normal;mso-bidi-font-style:normal'><span
style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>wide peak
boundaries</span></i><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'></p>

<p class=MsoNormal><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>These
rows identify the lesion in the same way as the all lesions file.</p>

<p class=MsoNormal><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>The
remaining rows list the genes contained in each wide peak.<span
style="mso-spacerun: yes">&nbsp; </span>For peaks that contain no genes, the
nearest gene is listed in brackets.</p>

<p class=MsoNormal><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal;mso-bidi-font-weight:
normal'><span lang=ES style='font-family:Arial;mso-bidi-font-family:"Times New Roman";
mso-ansi-language:ES'>3. Deletion Genes File </span></b><span lang=ES
style='font-family:Arial;mso-bidi-font-family:"Times New Roman";mso-ansi-language:
ES'>(del_genes.conf_<i style='mso-bidi-font-style:normal;mso-bidi-font-style:
normal'>XX</i>.txt, where <i style='mso-bidi-font-style:normal;mso-bidi-font-style:
normal'>XX</i> is the confidence level)</p>

<p class=MsoNormal><span lang=ES style='font-family:Arial;mso-bidi-font-family:
"Times New Roman";mso-ansi-language:ES'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>The
del genes file contains one column for each deletion peak identified in the
GISTIC analysis.<span style="mso-spacerun: yes">&nbsp; </span>The file format
for the del genes file is identical to the format for the amp genes file.</p>

<p class=MsoNormal><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='mso-bidi-font-family:"Times New Roman";
mso-no-proof:yes'><!--[if gte vml 1]><v:shape id="Picture_x0020_3" o:spid="_x0000_i1026"
 type="#_x0000_t75" alt="image004.jpg" style='width:580pt;height:83pt;
 visibility:visible;mso-wrap-style:square'>
 <v:imagedata src="GISTICDocumentation_standalone_files/image001.jpg" o:title="image004.jpg"/>
 <v:textbox style='mso-rotate-with-shape:t'/>
</v:shape><![endif]--><![if !vml]><img border=0 width=582 height=85
src="GISTICDocumentation_standalone_files/image002.png" alt=image004.jpg
v:shapes="Picture_x0020_3"><![endif]></span><span style='mso-bidi-font-family:
"Times New Roman"'></p>

<p class=MsoNormal><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal;mso-bidi-font-weight:
normal'><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>4.<span
style="mso-spacerun: yes">&nbsp; </span>Gistic Scores File </span></b><span
style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>(scores.gistic)</p>

<p class=MsoNormal><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>The
scores file lists the q-values [presented as -log10(q)], G-scores, average
amplitudes among aberrant samples, and frequency of aberration, across the
genome for both amplifications and deletions. The scores file is viewable with
the Genepattern SNPViewer module and may be imported into the Integrated
Genomics Viewer (IGV).</p>

<p class=MsoNormal><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='mso-bidi-font-family:"Times New Roman";
mso-no-proof:yes'><!--[if gte vml 1]><v:shape id="Picture_x0020_0" o:spid="_x0000_i1027"
 type="#_x0000_t75" alt="image006.jpg" style='width:477pt;height:149pt;
 visibility:visible;mso-wrap-style:square'>
 <v:imagedata src="GISTICDocumentation_standalone_files/image003.jpg" o:title="image006.jpg"/>
 <v:textbox style='mso-rotate-with-shape:t'/>
</v:shape><![endif]--><![if !vml]><img border=0 width=479 height=151
src="GISTICDocumentation_standalone_files/image004.png" alt=image006.jpg
v:shapes="Picture_x0020_0"><![endif]></span><span style='mso-bidi-font-family:
"Times New Roman"'></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal;mso-bidi-font-weight:
normal'><span style='mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></b></p>

<p class=MsoNormal><span style='mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal;mso-bidi-font-weight:
normal'><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>
5. Segmented Copy Number </span></b><span style='font-family:Arial;mso-bidi-font-family:
"Times New Roman"'>(raw_copy_number.pdf)<b style='mso-bidi-font-weight:normal;
mso-bidi-font-weight:normal'><o:p></o:p></b></span></p>

<p class=MsoNormal><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>
A .pdf or .png file containing a heatmap image of the
genomic profiles of the segmented input copy number data. The genome is represented
along the vertical axis and samples are arranged horizontally. 
<img  border=0 width=800 height=600 src="GISTICDocumentation_standalone_files/raw_copy_number.png">

</p>

<p class=MsoNormal><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal;mso-bidi-font-weight:
normal'><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></b></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal;mso-bidi-font-weight:
normal'><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>
6. Amplification Score GISTIC plot </span></b><span style='font-family:Arial;
mso-bidi-font-family:"Times New Roman"'>(amp_qplot.pdf)</p>

<p class=MsoNormal><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>
The amplification pdf is a plot of the G-scores (top) and q-values (bottom) with
respect to amplifications for all markers over the entire region analyzed.</p>

<img border=0 width=550 height=600 src="GISTICDocumentation_standalone_files/amp_qplot.png">

<p class=MsoNormal><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal;mso-bidi-font-weight:
normal'><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>
7. Deletion Score/q-value GISTIC plot</span></b><span style='mso-bidi-font-family:"Times New Roman"'>
</span><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>(del_qplot.pdf)</p>

<p class=MsoNormal><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'>
The deletion pdf is a plot of the G-scores (top) and q-values (bottom) with respect
to deletions for all markers over the entire region analyzed.
<img border=0 width=550 height=600 src="GISTICDocumentation_standalone_files/del_qplot.png">

</p>

<p class=MsoNormal><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal;mso-bidi-font-weight:
normal'><span style='font-family:Arial;mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></b></p>

</div>

</body>

</html>
