<!DOCTYPE html>
<html>
<head>
<title>GISTIC Documentation</title>
<style>
body {
  font-family: Arial;
}
p {
  font-family: Arial;
}
th {
  text-align: left;
  white-space: nowrap;
}
td {
  vertical-align: top;
}
.paramtable {
  padding: 2px 16px;
  font-size: 14px;
}
span {
  text-color: blue;
  text-align: right;
}
.infile { display:block }
.infile span { display:block;float:right;margin-left:10px;margin-right:10px;font-style:italic; }


</style>
</head>

<body>

<h2>GISTIC2 Documentation</h2>

<p>
<table>
<tr>
<td><b>Module Name:</b></td> <td>GISTIC2</td>
</tr>
<tr>
<td><b>Description:</b></td> <td>Genomic Identification of Significant Targets in Cancer, version 2.0</td>
</tr>
<tr>
<td><b>Authors:</b></td> <td>Gad Getz, Rameen Beroukhim, Craig Mermel, Steve Schumacher and Jen Dobson</td>
</tr> 
<tr>
<td><b>Date:</b></td> <td>27 Mar 2017</td>
</tr>
<tr>
<td><b>Release:</b></td> <td>2.0.23</td>
</tr>
</table>
<p>
<b>Summary:</b>

<p> 

The GISTIC module identifies regions of the genome that are significantly amplified or deleted across a set of samples.  Each aberration is assigned a G-score that considers the amplitude of the aberration as well as the frequency of its occurrence across samples.  False Discovery Rate q-values are then calculated for the aberrant regions, and regions with q-values below a user-defined threshold are considered significant.  For each significant region, a "peak region" is identified, which is the part of the aberrant region with greatest amplitude and frequency of alteration.  In addition, a "wide peak" is determined using a leave-one-out algorithm to allow for errors in the boundaries in a single sample. The "wide peak" boundaries are more robust for identifying the most likely gene targets in the region. Each significantly aberrant region is also tested to determine whether it results primarily from broad events (longer than half a chromosome arm), focal events, or significant levels of both.  The GISTIC module reports the genomic locations and calculated q-values for the aberrant regions.  It identifies the samples that exhibit each significant amplification or deletion, and it lists genes found in each "wide peak" region.

<p>


<b>Resources:</b>
<p>
<ul> 
  <li>
    <a href="https://github.com/broadinstitute/gistic2">Source code</a> on GitHub
  </li>
  <li>
    <a href="ftp://ftp.broadinstitute.org/pub/GISTIC2.0/">Linux x86-64 Executable tarball</a> via FTP at Broad Institute
  </li>
  <li>
    <a href="https://cloud.genepattern.org">Cloud implementation</a> at GenePattern
  </li>
</ul>
<p>
 
 <b>Publications:</b>
<p>
<ul> 
<li>
Mermel C, Schumacher S, et al. (2011). "<a href="https://pubmed.ncbi.nlm.nih.gov/21527027/">GISTIC2.0 facilitates sensitive and confident localization of the targets of focal somatic copy-number alteration in human cancers.</a>" <i>Genome Biology</i>, <b>12</b>:R41.
</li>
<li>
Beroukhim R, Mermel C, et al. (2010). "<a href="https://pubmed.ncbi.nlm.nih.gov/20164920/">The landscape of somatic copy-number alteration across human cancers.</a>" <i>Nature</i>, <b>463</b>:899-905.
</li>
<li>
Beroukhim R, Getz G, et al. (2007).  "<a href="https://pubmed.ncbi.nlm.nih.gov/18077431/">Assessing the significance of chromosomal abberations in cancer: Methodology and application to glioma.</a>"  <i>Proc Natl Acad Sci</i>, <b>104</b>:20007-20012.
</li>
</ul>
<p>
  
<h3>Input Parameters</h3>
<div class="paramtable">
<i>Variable Name</i> refers to the input parameter when running GISTIC by calling it as a MATLAB<sup>TM</sup> function; <i>Option String</i> refers to the command-line parameter when running GISTIC as a stand-alone executable program.
<p>
<table>
<tr>
<th>Variable Name</th>
<th>Option String&nbsp;&nbsp;&nbsp;&nbsp;</th>
<th>Description</th>
</tr>
<tr>
<td><i>base_dir</i></td>
<td>-b</td>
<td>The directory in which to save all output files. (<em>REQUIRED</em>)</td>
</tr>
<tr>
<td><i>seg_file</i></td>
<td>-seg</td>
<td>Path to segmented data input file (<em>REQUIRED</em>, see below for file description).</td>
</tr>
<tr>
<td><i>refgene</i></td>
<td>-refgene</td>
<td>Path to reference genome data input file (<em>REQUIRED</em>, see below for file description).</td>
</tr>
<tr>
<td><i>markers</i> (file path)</td>
<td>-mk</td>
<td>Path to markers input file (<em>OPTIONAL</em>, but encouraged: see below for file description).</td>
</tr>
<tr>
<td><i>markers</i> (numeric)</td>
<td>-maxspace</td>
<td>Maximum allowed spacing between pseudo-markers, in bases. Pseudo-markers are generated when the markers file input is omitted.  Segments that contain fewer than this number of markers are joined to the neighboring segment that is closest in copy number.  (DEFAULT=10,000)</td>
</tr>
<tr>
<td><i>t_amp</i></td>
<td>-ta</td>
<td>Threshold for copy number amplifications.  Regions with a copy number gain above this positive value are considered amplified. Regions with a copy number gain smaller than this value are considered noise and set to 0. (DEFAULT=0.1)</td>
</tr>
<tr>
<td><i>t_del</i></td>
<td>-td</td>
<td>Threshold for copy number deletions.  Regions with a copy number loss below the negative of this positive value are considered deletions. Regions with a smaller copy number loss are considered noise and set to 0. (DEFAULT=0.1)</td>
</tr>
<tr>
<td><i>join_segment_size</i></td>
<td>-js</td>
<td>Smallest number of markers to allow in segments from the segmented data.  Segments that contain fewer than this number of markers are joined to the neighboring segment that is closest in copy number.  (DEFAULT=4)</td>
</tr>
<tr>
<td><i>qv_thresh</i></td>
<td>-qvt</td>
<td>Significance threshold for q-values.  Regions with q-values below this number are considered significant. (DEFAULT=0.25)</td>
</tr>
<tr>
<td><i>ext</i></td>
<td>-ext</td>
<td>Extension to append to all output files.  (DEFAULT='', no extension)</td>
</tr>
<tr>
<td><i>fname</i></td>
<td>-fname</td>
<td>Base file name to prepend to all output files.  (DEFAULT no output basename)</td>
</tr>
<tr>
<td><i>remove_X</i></td>
<td>-rx</td>
<td>Flag indicating whether to remove data from the sex chromosomes before analysis.  Allowed values= {1,0}. (DEFAULT=1, remove X,Y)</td>
</tr>
<tr>
<td><i>cap</i></td>
<td>-cap</td>
<td>Minimum and maximum cap values on analyzed data. Regions with a log2 ratio greater than the cap are set to the cap value; regions with a log2 ratio less than -cap value are set to -cap. Values must be positive. (DEFAULT=1.5)</td>
</tr>
<tr>
<td><i>run_broad_analysis</i></td>
<td>-broad</td>
<td>Flag indicating that an additional broad-level analysis should be performed. Allowed values = {1,0}. (DEFAULT = 0, no broad analysis).</td>
</tr>
<tr>
<td><i>broad_len_cutoff</i></td>
<td>-brlen</td>
<td>Threshold used to distinguish broad from focal events, given in units of fraction of chromosome arm. (DEFAULT = 0.98)</td>
</tr>
<tr>
<td><i>use_two_sided</i></td>
<td>-twoside</td>
<td>Flag indicating that a two-dimensional quadrant figure should be created as part of a broad analysis. Allowed values = {1,0}.(DEFAULT=0, no figure).</td>
</tr>
<tr>
<td><i>ziggs.max_segs_per_sample</i></td>
<td>-maxseg</td>
<td>Maximum number of segments allowed for a sample in the input data. Samples with more segments than this threshold are excluded from the analysis. (DEFAULT=2500)</td>
</tr>
<tr>
<td><i>res</i></td>
<td>-res</td>
<td>Resolution used to create the empirical distributions used to estimate background probabilities. Lower values generate more accurate results at a cost of greater computation time. (DEFAULT=0.05)</td>
</tr>
<tr>
<td><i>conf_level</i></td>
<td>-conf</td>
<td>Confidence level used to calculate the region containing a driver. (DEFAULT=0.75)</td>
</tr>
<tr>
<td><i>do_gene_gistic</i></td>
<td>-genegistic</td>
<td>Flag indicating that the gene GISTIC algorithm should be used to calculate the significance of deletions at a gene level instead of a marker level. Allowed values= {1,0}. (DEFAULT=0, no gene GISTIC).</td>
</tr>
<tr>
<td><i>do_arbitration</i></td>
<td>-arb</td>
<td>Flag for using the arbitrated peel-off algorithm when resolving the significance of overlapping peaks. Allowed values= {1,0}. (DEFAULT=1, use arbitrated peel-off)</td>
</tr>
<tr>
<td>peak_types</td>
<td>-peak_type</td>
<td>Method for evaluating the significance of peaks, either robust (DEFAULT) or loo for leave-one-out.</td>
</tr>
<tr>
<td><i>arm_peeloff</i></td>
<td>-armpeel</td>
<td>Flag set to enable arm-level peel-off of events during peak definition. The arm-level peel-off enhancement to the arbitrated peel-off method assigns all events in the same chromosome arm of the same sample to a single peak. It is useful when peaks are split by noise or chromothripsis. Allowed values= {1,0}. (DEFAULT=0, use normal arbitrated peel-off)</td>
</tr>
<tr>
<td><i>sample_center</i></td>
<td>-scent</td>
<td>Method for centering each sample prior to the GISTIC analysis. Allowed values are median, mean, or none. (DEFAULT=median)</td>
</tr>
<tr>
<td><i>conserve_disk_space</i></td>
<td>-smalldisk</td>
<td>Flag indicating that large MATLAB<sup>TM</sup> objects should not be saved to disk. Allowed values= {1,0}. (DEFAULT=0, save large).</td>
</tr>
<tr>
<td><i>use_segarray</i></td>
<td>-smallmem</td>
<td>Flag indicating that the SegArray memory compression scheme should be used to reduce the memory requirements of the computation for large data sets. Computation is somewhat slower with memory compression enabled. Allowed values= {1,0}. (DEFAULT=1, compress memory)</td>
</tr>
<tr>
<td><i>write_gene_files</i></td>
<td>-savegene</td>
<td>Flag indicating that gene tables should be saved. Allowed values= {1,0}. (DEFAULT=0, don't save gene tables)</td>
</tr>
<tr>
<td><i>gene_collapse_method</i></td>
<td>-gcm</td>
<td>Method for reducing marker-level copy number data to the gene-level copy number data in the gene tables. Markers contained in the gene are used when available, otherwise the flanking marker or markers are used. Allowed values are mean, median, min, max or extreme. The extreme method chooses whichever of min or max is furthest from diploid. (DEFAULT=mean)</td>
</tr>
<tr>
<td><i>save_seg_data</i></td>
<td>-saveseg</td>
<td>Flag indicating that the preprocessed segmented data used as input for the GISTIC analysis should be saved (in matlab format). Allowed values= {1,0}. (DEFAULT=1, save segmented input data)</td>
</tr>
<tr>
<td><i>save_data_files</i></td>
<td>-savedata</td>
<td>Flag indicating that native MatlabTM output files should be saved in addition to text data. Allowed values= {1,0}. (DEFAULT=1, save MatlabTM files)</td>
</tr>
<tr>
<td><i>use set_verbose_level()</i></td>
<td>-v</td>
<td>Integer value indicating the level of verbosity to use in the program execution log.  Suggested values = {0,10,20,30}.  0 sets no verbosity; 30 sets high level of verbosity.  (DEFAULT=0)</td>
</tr>
</table>
</div>
<p>

<h3>Input Files</h3>
<p>

<ol class="infile">
 
<li><b>Segmentation File</b> (-seg) <span>REQUIRED</span>
<p>
The segmentation file contains the segmented data for all the samples identified by GLAD, CBS, or some other segmentation algorithm.  (See GLAD file format in the Genepattern file formats documentation.)  It is a six column, tab-delimited file with an optional first line identifying the columns.  Positions are in base pair units.
<p>
 The column headers are:
<p>

<ul style="list-style-type:none;">
<li>(1)  <i>Sample</i> &nbsp;&nbsp;&nbsp;&nbsp; (sample name)</li>
<li>(2)  <i>Chromosome</i> &nbsp;&nbsp;&nbsp;&nbsp; (chromosome number)</li>
<li>(3)  <i>Start Position</i> &nbsp;&nbsp;&nbsp;&nbsp; (segment start position, in bases)</li>
<li>(4)  <i>End Position</i> &nbsp;&nbsp;&nbsp;&nbsp; (segment end position, in bases)</li>
<li>(5)  <i>Num Markers</i> &nbsp;&nbsp;&nbsp;&nbsp; (number of markers in segment)</li>
<li>(6)  <i>Seg.CN</i> &nbsp;&nbsp;&nbsp;&nbsp; (log2() -1 of copy number)</li>
</ul>

<p> 
<a href="examplefiles/segmentationfile.txt" target="_blank">Example Segmentation File</a>
<p>
</li>

<li><b>Markers File</b> (-mk) <span>OPTIONAL</span>
<p>
The markers file identifies the marker positions used in the original dataset (before segmentation) for array or capture experiments. As of GISTIC release 2.0.23, the markers file is an optional input - if omitted, pseudo markers are generated as uniformly as possible using the maxspace input parameter.  
<p>

The markers file is a three column, tab-delimited file with an optional header.  The column headers are:
<p>
<ul style="list-style-type:none;">
<li>(1)  <i>Marker Name</i></li>
<li>(2)  <i>Chromosome</i></li>
<li>(3)  <i>Marker Position</i>&nbsp;&nbsp;&nbsp;&nbsp;(in bases)</li>
</ul>
<p>

<a href="examplefiles/markersfile.txt" target="_blank">Example Markers File</a>
</li>

<li><b>Reference Genome File</b> (-refgene) <span>REQUIRED</span>
<p>
The reference genome file contains information about the location of genes and cytobands on a given build of the genome. Reference genome files are created in MatlabTM and are not viewable with a text editor. The GISTIC 2.0 release has four reference genomes located in the refgenefiles directory: hg16.mat, hg17.mat, hg18.mat, and hg19.mat. 
</li>

<li><b>Array List File</b> (-alf) <span>OPTIONAL</span>
<p>
The array list file is an optional file identifying the subset of samples to be used in the analysis.  It is a one column file with an optional header (array).  The sample identifiers listed in the array list file must match the sample names given in the segmentation file. 
<p>
<a href="examplefiles/arraylistfile.txt" target="_blank">Example Array List File</a>
</li>

<li><b>CNV File</b> (-cnv) <span>OPTIONAL</span>
<p>
There are two options for the file specifying germ line CNVs to be excluded from the analysis.  The first option allows CNVs to be identified by marker name and is platform-specific.  The second option allows the CNVs to be identified by genomic location, which is platform independent but genome-build dependent.
<p>
<u>Option #1</u>:  A two column, tab-delimited file with an optional header row.  The marker names given in this file must match the marker names given in the markers file.  The CNV identifiers are for user use and can be arbitrary.  The column headers are:
<p>
<ul style="list-style-type:none;">
<li>(1)  <i>Marker Name</i></li>
<li>(2)  <i>CNV Identifier</i></li>
<p>
</ul>
<u>Option #2</u>:  A 6 column, tab-delimited file with an optional header row.  The 'CNV Identifier' is for user use and can be arbitrary.  'Narrow Region Start' and 'Narrow Region End' are also not used.  The column headers are:
<p>
<ul style="list-style-type:none;">
<li>(1)  <i>CNV Identifier</i></li>
<li>(2)  <i>Chromosome</i></li>
<li>(3)  <i>Narrow Region Start</i></li>
<li>(4)  <i>Narrow Region End</i></li>
<li>(5)  <i>Wide Region Start</i></li>
<li>(6)  <i>Wide Region End</i></li>
</ul>
 
<p>
<a href="examplefiles/cnvfile.txt" target="_blank">Example CNV File</a>
<p>
</li>

</ol> 

<h3>Output Files</h3>
<p>

<ol> 
<li><b>All Lesions File</b> (all_lesions.conf_<i>XX</i>.txt, where <i>XX</i> is the confidence level)
<p>
The all lesions file summarizes the results from the GISTIC run.  It contains data about the significant regions of amplification and deletion as well as which samples are amplified or deleted in each of these regions.  The identified regions are listed down the first column, and the samples are listed across the first row, starting in column 10. 
<p>

<em>Region Data</em>
<p>
Columns 1-9 present the data about the significant regions as follows:
<ul style="list-style-type:none;">
<li>(1)  <i>Unique Name:</i> A name assigned to identify the region</li>
<li>(2)  <i>Descriptor:</i> The genomic descriptor of that region.</li>
<li>(3)  <i>Wide Peak Limits:</i> The "wide peak" boundaries most likely to contain the targeted genes.  These are listed in genomic coordinates and marker (or probe) indices.</li>
<li>(4)  <i>Peak Limits:</i> The boundaries of the region of maximal amplification or deletion.</li>
<li>(5)  <i>Region Limits:</i> The boundaries of the entire significant region of amplification or deletion.</li>
<li>(6)  <i>q-values:</i> The q-value of the peak region.</li>
<li>(7)  <i>Residual q-values:</i> The q-value of the peak region after removing ("peeling off") amplifications or deletions that overlap other, more significant peak regions in the same chromosome.</li>
<li>(8)  <i>Broad or Focal:</i> Identifies whether the region reaches significance due primarily to broad events (called "broad"), focal events (called "focal"), or independently significant broad and focal events (called "both").</li>
<li>(9)  <i>Amplitude Threshold:</i> Key giving the meaning of values in the subsequent columns associated with each sample.</li>
</ul>
<p>
<em>Sample Data</em>
<p>
Each of the analyzed samples is represented in one of the columns following the lesion data (columns 10 through end).  The data contained in these columns varies slightly by section of the file. 
<p>
The first section can be identified by the key given in column 9 – it starts in row 2 and continues until the row that reads "Actual Copy Change Given."  This section contains summarized data for each sample.  A '0' indicates that the copy number of the sample was not amplified or deleted beyond the threshold amount in that peak region.  A '1' indicates that the sample had low-level copy number aberrations (exceeding the low threshold indicated in column 9), and a '2' indicates that the sample had high-level copy number aberrations (exceeding the high threshold indicated in column 9).
<p>
The second section can be identified the rows in which column 9 reads "Actual Copy Change Given."  The second section exactly reproduces the first section, except that here the actual changes in copy number are provided rather than zeroes, ones, and twos.
<p>
The final section is similar to the first section, except that here only broad events are included.  A 1 in the samples columns (columns 10+) indicates that the median copy number of the sample across the entire significant region exceeded the threshold given in column 9.  That is, it indicates whether the sample had a geographically extended event, rather than a focal amplification or deletion covering little more than the peak region.
<p>
</li>

<li><b>Amplification Genes File</b> (amp_genes.conf_<i>XX</i>.txt, where <i>XX</i> is the confidence level)
<p>
The amp genes file contains one column for each amplification peak identified in the GISTIC analysis.  The first four rows are:
<p>
<ul style="list-style-type:none;">
<li>(1) <i>cytoband</i></li>
<li>(2) <i>q-value</i></li>
<li>(3) <i>residual q-value</i></li>
<li>(4) <i>wide peak boundaries</i></li>
</ul>
<p>
These rows identify the lesion in the same way as the all lesions file.
<p>
The remaining rows list the genes contained in each wide peak.  For peaks that contain no genes, the nearest gene is listed in brackets.
<p>
</li>

<li><b>Deletion Genes File</b> (del_genes.conf_<i>XX</i>.txt, where <i>XX</i> is the confidence level)
<p>
The del genes file contains one column for each deletion peak identified in the GISTIC analysis.  The file format for the del genes file is identical to the format for the amp genes file.
<div>
<img border=0 width=582 height=85 src="images/image002.png"
</div>
</li>

<li><b>Gistic Scores File</b> (scores.gistic)
<p>
The scores file lists the q-values [presented as -log10(q)], G-scores, average amplitudes among aberrant samples, and frequency of aberration, across the genome for both amplifications and deletions. The scores file is viewable with the Genepattern SNPViewer module and may be imported into the Integrated Genomics Viewer (IGV).
<div>
<img border=0 width=479 height=151 src="images/image004.png">
</div>
</li>

<li><b>Segmented Copy Number</b> (raw_copy_number.pdf)
<p>
A .pdf or .png file containing a heatmap image of the genomic profiles of the segmented input copy number data. The genome is represented along the vertical axis and samples are arranged horizontally.
<div>
<img  border=0 width=800 height=600 src="images/raw_copy_number.png">
</div>
</li>

<li><b>Amplification Score GISTIC plot</b> (amp_qplot.pdf)
<p>
The amplification pdf is a plot of the G-scores (top) and q-values (bottom) with respect to amplifications for all markers over the entire region analyzed.
<div>
<img border=0 width=550 height=600 src="images/amp_qplot.png">
</div>
</li>

<li><b>Deletion Score/q-value GISTIC plot</b> (del_qplot.pdf)
<p>
The deletion pdf is a plot of the G-scores (top) and q-values (bottom) with respect to deletions for all markers over the entire region analyzed.
<div>
<img border=0 width=550 height=600 src="images/del_qplot.png">
</div>
</li>
</ol>
 

</body>
</html>
